---
title: "Zebrafish Diet and Infection Targetted Analysis"
author: "Michael Sieler"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: yes
    toc_depth: 3
    number_sections: yes
    theme: united
    highlight: tango
  pdf_document:
    toc: yes
    toc_depth: '3'
---



```{r setup, include=FALSE}


# Setup environment

knitr::opts_chunk$set(
	message = FALSE,
	warning = FALSE
)
  # Sets default knit settings for document
Sys.Date()
analysisHelper.Date = "2022-02-28"  # "2022-02-30"
rerun.analysesHelper = F # F = Do not rerun lengthy functions, T = Rerun lengthy functions
rerun.DiffAbund = F  # F = Do not rerun lengthy diff abund functions, T = Rerun Diff Abund functions

# Subsets alpha plots, to only display one metric
plot.oneAlpha <- F
# if(isTRUE(plot.oneAlpha)){
#   subset.alphaMetric <- "Shannon"
# }
# if(isTRUE(plot.oneAlpha)){
#   data <- subset(data$Alpha.Metric == subset.alphaMetric)
# }



# Check project file location is in correct directory
## Path to project directory (Rproject file should be stored in "~/Code")
if(grepl("/Code", getwd())){
  proj.path <- getwd()
} else(
  # Script will stop if project file is not located in ~/Code directory
  stop("Project file (`.Rproj`) is not located in `~/Code` directory. 
       \t**Exiting script.")
)


# Helper Scripts

## Load libraries
source(paste0(proj.path,"/HelperScripts/libraries.R"))

## Load functions
source(paste0(proj.path,"/HelperScripts/functions_2.R"))

## Load custom plot settings
source(paste0(proj.path,"/HelperScripts/plot_settings.R"))


# Set important paths

## Path to data files/sub-folders
data.path <- check_proj_struct(proj.path,  # Project path
                               "/../Data")  # Sub-directory

## Path to Robjects (Renv, Rds, Rdata, etc.) saved here
objects.path <- check_proj_struct(data.path,  # Project path
                                  "/Clean/Robjects")  # Sub-directory

## Path to Output (Figures, Tables)
output.path <- check_proj_struct(proj.path,  # Project path
                                 "/../Output")  # Sub-directory

## Load Config file (Deprecated)
# source(paste0(proj.path,"/Options/options.R"))


## Load Analysis Helper or Rerun
if(isTRUE(rerun.analysesHelper)) {
  
  # Run Pre-analysis helper file
  source(paste0(proj.path,"/HelperScripts/analysis-helper.R"))
  save.image(paste0(objects.path, "/analysis-helper-objects-and-env_ZF-DietStudy_", Sys.Date() ,"_ENV.RData"))
} else {
  
  # Load 
  load(file.path(paste0(objects.path, "/analysis-helper-objects-and-env_ZF-DietStudy_", analysisHelper.Date,"_ENV.RData")))
}




# Save Environment
# save_env(objects.path, ID = analysis.ID, extra_info = "_finishedAnalysis_output")
# save.image("~/Dropbox/Mac (2)/Documents/Project_Repos/Plug-n-Chug-Microbiome-16S-Data-Analysis/Data/Clean/Robjects/analysis-helper-objects-and-env_ZF-DietStudy_2022-02-08_ENV.RData")

```



# Diet by endpoint analyses

## Weight

### Does diet affect physiology (weight)?

```{r Weight-Diet, message=FALSE, warning=FALSE, include=FALSE}

data <- df.T1

# lm()
mod  <- lm( data = data, Weight ~ Diet )
formula = "Weight ~ Diet"

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Weight",  # y-variable
                      # facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = formula,  # Optional
                      # title = NULL,  # Optional.
                      # caption = "0 = uninfected, 1 = infected"  # Optional
                      y.lab = "Weight (mg)",
                      comparisons = list( c("Gemma", "Watts"), 
                                          c("Watts", "ZIRC"), 
                                          c("Gemma", "ZIRC") )
                      )  


# p1 <- ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Weight,
#   # grouping.var     = genre, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
#   # type = ""
# )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# p1

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)


# disp_stats(p1)


```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
mod %>% report()
# mod %>% report_table()
```

\newpage


### Does sex affect physiology (weight)?
* Weight ~ sex


```{r Weight-sex, message=FALSE, warning=FALSE, include=FALSE}

data <- df.T1

# lm()
mod  <- lm( data = data, Weight ~ Sex )
formula = "Weight ~ Sex"

# Plot
plot <- gen_box_plot_2(data = data,
                      "Sex",  # x-variable
                      "Weight",  # y-variable
                      # facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = formula  # Optional
                      # title = NULL,  # Optional.
                      # caption = "0 = uninfected, 1 = infected"  # Optional
                      )  


# p1 <- ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Weight,
#   # grouping.var     = genre, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
#   # type = ""
# )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# p1

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)


# disp_stats(p1)


```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
mod %>% report()
# mod %>% report_table()
```




\newpage


### Does diet affect physiology (weight) depending on sex?
* Weight ~ diet*sex


```{r Weight-Diet-Sex, message=FALSE, warning=FALSE, include=FALSE}

data <- df.T1
formula = "Weight ~ Diet*Sex"
# lm()
mod  <- lm( data = data, Weight ~ Diet*Sex )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = formula  # Optional
                      # title = NULL,  # Optional.
                      # caption = "0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Sex",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = formula  # Optional
                      # title = NULL,  # Optional.
                      # caption = "0 = uninfected, 1 = infected"  # Optional
                      )

p1 <- grouped_ggbetweenstats(
  data             = data,
  x                = Diet,
  y                = Weight,
  grouping.var     = Sex, ## grouping variable
  outlier.tagging  = TRUE, ## whether outliers need to be tagged
  # outlier.label    = title, ## variable to be used for tagging outliers
  outlier.coef     = 2,
  ggsignif.args    = list(textsize = 4, tip_length = 0.01),
  p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
  ## adding new components to `{ggstatsplot}` default
  ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
  # caption          = "Source: IMDb (Internet Movie Database)",
  palette          = "Dark2",
  package          = "RColorBrewer",
  plotgrid.args    = list(nrow = 1),
  # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
)

# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Sex,
#   y                = Weight,
#   grouping.var     = Diet, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 2),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p2)


```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

### Does diet affect physiology (weight) depending on time point?
* Weight ~ diet*timepoint


```{r Weight-Diet-Time, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
formula <- "Weight ~ Diet*Timepoint"

# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = formula  # Optional.
                      # caption = formula #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = formula  # Optional.
                      # caption = formula #"0 = uninfected, 1 = infected"  # Optional
                      )

# p1 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Weight,
#   grouping.var     = Timepoint, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
# 
# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Timepoint,
#   y                = Weight,
#   grouping.var     = Diet, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

### Does sex affect physiology (weight) depending on time point?
* Weight ~ diet*timepoint


```{r Weight-sex-Time, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
formula <- "Weight ~ Sex*Timepoint"

# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Sex",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = formula  # Optional.
                      # caption = formula #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = formula  # Optional.
                      # caption = formula #"0 = uninfected, 1 = infected"  # Optional
                      )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
mod %>% report()
# mod %>% report_table()
```

\newpage

### Does diet affect physiology (weight) depending on time point in males?
* Weight ~ diet*timepoint


```{r Weight-Diet-Time-Males, message=FALSE, warning=FALSE, include=FALSE}

data <- subset(df.conT0T1, Sex == "M")
formula <- "Weight ~ Diet*Timepoint"

# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = formula,  # Optional.
                      caption = "Only males" #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = formula,  # Optional.
                      caption = "Only males" #"0 = uninfected, 1 = infected"  # Optional
                      )

# p1 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Weight,
#   grouping.var     = Timepoint, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
# 
# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Timepoint,
#   y                = Weight,
#   grouping.var     = Diet, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2


# ggcoefstats(mod) 


# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

### Does time affect physiology (weight) depending on sex?
* Weight ~ diet*timepoint


```{r Weight-Time-Sex, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
formula <- "Weight ~ Timepoint*Sex"

# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = formula,  # Optional.
                      caption = "Only males" #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Sex",  # x-variable
                      "Weight",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = formula,  # Optional.
                      caption = "Only males" #"0 = uninfected, 1 = infected"  # Optional
                      )

# p1 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Timepoint,
#   y                = Weight,
#   grouping.var     = Sex, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
# 
# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Sex,
#   y                = Weight,
#   grouping.var     = Timepoint, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2


# ggcoefstats(mod) 


# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

## Body Condition Score

### Does diet affect physiology (body condition score)?
* Body.Condition.Score ~ diet


```{r BodyCond-Diet, message=FALSE, warning=FALSE, include=FALSE}

data <- df.T1
formula <- "Body.Condition.Score ~ Diet"
title <- "Body.Condition.Score ~ Diet"

# lm()
mod  <- lm( data = data, Body.Condition.Score ~ Diet )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      # facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = title  # Optional
                      # title = NULL,  # Optional.
                      # caption = "0 = uninfected, 1 = infected"  # Optional
                      )  


# p1 <- ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Body.Condition.Score,
#   # grouping.var     = genre, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
#   # type = ""
# )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# p1

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)


# disp_stats(p1)


```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
mod %>% report()
# mod %>% report_table()
```

\newpage


### Does sex affect physiology (body condition score)?
* Body.Condition.Score ~ sex


```{r BodyCond-sex, message=FALSE, warning=FALSE, include=FALSE}

data <- df.T1
formula <- "Body.Condition.Score ~ Sex"
title <- "Body.Condition.Score ~ Sex"

# lm()
mod  <- lm( data = data, Body.Condition.Score ~ Sex )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Sex",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      # facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      # title = NULL,  # Optional.
                      title = title
                      # caption = "0 = uninfected, 1 = infected"  # Optional
                      )  


# p1 <- ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Body.Condition.Score,
#   # grouping.var     = genre, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
#   # type = ""
# )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# p1

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)


# disp_stats(p1)


```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
mod %>% report()
# mod %>% report_table()
```




\newpage


### Does diet affect physiology (body condition score) depending on sex?
* Body.Condition.Score ~ diet*sex


```{r BodyCond-Diet-Sex, message=FALSE, warning=FALSE, include=FALSE}

data <- df.T1
formula <- "Body.Condition.Score ~ Diet*Sex"
title <- "Body.Condition.Score ~ Diet*Sex"

# lm()
mod  <- lm( data = data, Body.Condition.Score ~ Diet*Sex )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title
                      # title = NULL,  # Optional.
                      # caption = "0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Sex",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = title  # Optional
                      # title = NULL,  # Optional.
                      # caption = "0 = uninfected, 1 = infected"  # Optional
                      )

# p1 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Body.Condition.Score,
#   grouping.var     = Sex, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )

# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Sex,
#   y                = Body.Condition.Score,
#   grouping.var     = Diet, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 2),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p2)


```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

### Does diet affect physiology (body condition score) depending on time point?
* Body.Condition.Score ~ diet*timepoint


```{r BodyCond-Diet-Time, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
formula <- "Body.Condition.Score ~ Diet*Timepoint"
title <- "Body.Condition.Score ~ Diet*Timepoint"

# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title  # Optional.
                      # caption = formula #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title  # Optional.
                      # caption = formula #"0 = uninfected, 1 = infected"  # Optional
                      )

# p1 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Body.Condition.Score,
#   grouping.var     = Timepoint, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
# 
# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Timepoint,
#   y                = Body.Condition.Score,
#   grouping.var     = Diet, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

### Does sex affect physiology (body condition score) depending on time point?
* Body.Condition.Score ~ sex*timepoint


```{r BodyCond-sex-Time, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
formula <- "Body.Condition.Score ~ Sex*Timepoint"
title <- "Body.Condition.Score ~ Sex*Timepoint"

# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Sex",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title  # Optional.
                      # caption = formula #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title  # Optional.
                      # caption = formula #"0 = uninfected, 1 = infected"  # Optional
                      )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2

# ggcoefstats(mod) 

# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
mod %>% report()
# mod %>% report_table()
```

\newpage

### Does diet affect physiology (body condition score) depending on time point in males?
* Body.Condition.Score ~ diet*timepoint


```{r BodyCond-Diet-Time-Males, message=FALSE, warning=FALSE, include=FALSE}

data <- subset(df.conT0T1, Sex == "M")
formula <- "Body.Condition.Score ~ Diet*Timepoint"
title <- "Body.Condition.Score ~ Diet*Timepoint"

# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title,  # Optional.
                      caption = "Only males" #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title,  # Optional.
                      caption = "Only males" #"0 = uninfected, 1 = infected"  # Optional
                      )

# p1 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Body.Condition.Score,
#   grouping.var     = Timepoint, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
# 
# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Timepoint,
#   y                = Body.Condition.Score,
#   grouping.var     = Diet, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2


# ggcoefstats(mod) 


# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

### Does diet affect physiology (body condition score) depending on time point in females?
* Body.Condition.Score ~ diet*timepoint


```{r BodyCond-Diet-Time-females, message=FALSE, warning=FALSE, include=FALSE}

data <- subset(df.conT0T1, Sex == "F")
formula <- "Body.Condition.Score ~ Diet*Timepoint"
title <- "Body.Condition.Score ~ Diet*Timepoint"

# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Diet",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      colors = col.diet,  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title,  # Optional.
                      caption = "Only females" #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      colors = col.sex,  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title,  # Optional.
                      caption = "Only females" #"0 = uninfected, 1 = infected"  # Optional
                      )

# p1 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Diet,
#   y                = Body.Condition.Score,
#   grouping.var     = Timepoint, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
# 
# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Timepoint,
#   y                = Body.Condition.Score,
#   grouping.var     = Diet, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2


# ggcoefstats(mod) 


# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

### Does time affect physiology (body condition score) depending on sex?
* Body.Condition.Score ~ diet*timepoint


```{r BodyCond-Time-Sex, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
formula <- "Body.Condition.Score ~ Timepoint*Sex"
title <- "Body.Condition.Score ~ Timepoint*Sex"
# lm()
mod  <- lm( data = data, formula = as.formula(formula) )

# Plot
plot <- gen_box_plot_2(data = data,
                      "Timepoint",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title,  # Optional.
                      caption = "Only males" #"0 = uninfected, 1 = infected"  # Optional
                      )

plot2 <- gen_box_plot_2(data = data,
                      "Sex",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = c("Timepoint"),  # Optional: c("Diet", "Sex") or "Diet"
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      # title = "",  # Optional
                      title = title,  # Optional.
                      caption = "Only males" #"0 = uninfected, 1 = infected"  # Optional
                      )

# p1 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Timepoint,
#   y                = Body.Condition.Score,
#   grouping.var     = Sex, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )
# 
# p2 <- grouped_ggbetweenstats(
#   data             = data,
#   x                = Sex,
#   y                = Body.Condition.Score,
#   grouping.var     = Timepoint, ## grouping variable
#   outlier.tagging  = TRUE, ## whether outliers need to be tagged
#   # outlier.label    = title, ## variable to be used for tagging outliers
#   outlier.coef     = 2,
#   ggsignif.args    = list(textsize = 4, tip_length = 0.01),
#   p.adjust.method  = "bonferroni", ## method for adjusting p-values for multiple comparisons
#   ## adding new components to `{ggstatsplot}` default
#   ggplot.component = list(ggplot2::scale_y_continuous(sec.axis = ggplot2::dup_axis())),
#   # caption          = "Source: IMDb (Internet Movie Database)",
#   palette          = "Dark2",
#   package          = "RColorBrewer",
#   plotgrid.args    = list(nrow = 1),
#   # annotation.args  = list(title = "Differences in movie length by mpaa ratings for different genres")
# )

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# p1
# p2


# ggcoefstats(mod) 


# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)
```

```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# mod %>% report()
# mod %>% report_table()
```

\newpage

## Infection, Severity Score

### Does diet affect infection status?
* infection.status ~ diet

```{r Infection-Diet, message=FALSE, warning=FALSE, include=FALSE}
# GLM
# mod <- gen_glm(data = dt.expFin,
#               x.var = "Diet", # singular or multiple variables as a list. Must be NULL if loop var is the only X-var. Default is NULL
#               y.var = "factor(Infection.Status)",
#               # loop = "x",  # Which variable to loop? "x" or "y", NA is default
#               # loop.var = methods.alpha, # List of variables to loop on
#               # extra.data = NA,  # need to merge dataframes on a sample or ID name? Enter datatable
#               # samp.name="Sample"  # Which columns do you want to merge on? Need to be the same (eg. "Sample")
#               terms = 1,  # How many interaction terms are you using? Default is 1
#               glm.fam = "binomial",  # What GLM family? Default is "gaussian"
#               # rand.eff = NA  # Random effect? Enter which variable you want to be a random effect
#               # fact.y = T
#               run.anv = F
#               )

data <- subset(dt.expFin)
formula = "Infection.Status ~ Diet"
title = "Infection.Status ~ Diet"

mod <- glm( factor(Infection.Status) ~ Diet,
            data = data,
            family = binomial(link="logit"))

# Plot
plot <- gen_bar_plot(data = data,
                              "Infection.Status",  # x-variable
                              # "Diet",  # y-variable
                              facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                              # facet.var.y = "Sex",  # Optional
                              # colors = "Sex",  # Optional, col.diet or "Sex"
                              # title = "",  # Optional
                              title = title,
                              # title = NULL,  # Optional.
                              caption = "0 = uninfected, 1 = infected"  # Optional
                              )
## for reproducibility
# p1 <- grouped_ggbarstats(
#   data         = df.expFin,
#   x            = Infection.Status,
#   y            = Sex,
#   type = "np",
#   grouping.var = Diet,
#   package      = "RColorBrewer",
#   palette      = "Dark2"
# )

# mod %>% report()

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# p1


# ggcoefstats(mod) 


# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)

```

\newpage

### Does diet affect infection status depending on sex?
* infection.status ~ diet

```{r Infection-Diet-sex, message=FALSE, warning=FALSE, include=FALSE}
# GLM
# mod <- gen_glm(data = dt.expFin,
#               x.var = "Diet", # singular or multiple variables as a list. Must be NULL if loop var is the only X-var. Default is NULL
#               y.var = "factor(Infection.Status)",
#               # loop = "x",  # Which variable to loop? "x" or "y", NA is default
#               # loop.var = methods.alpha, # List of variables to loop on
#               # extra.data = NA,  # need to merge dataframes on a sample or ID name? Enter datatable
#               # samp.name="Sample"  # Which columns do you want to merge on? Need to be the same (eg. "Sample")
#               terms = 1,  # How many interaction terms are you using? Default is 1
#               glm.fam = "binomial",  # What GLM family? Default is "gaussian"
#               # rand.eff = NA  # Random effect? Enter which variable you want to be a random effect
#               # fact.y = T
#               run.anv = F
#               )

data <- dt.expFin
formula = "Infection.Status ~ Diet*Sex"
title = "Infection.Status ~ Diet*Sex"

mod <- glm( factor(Infection.Status) ~ Diet*Sex,
            data = data,
            family = binomial(link="logit"))

# Plot
plot <- gen_bar_plot(data = data,
                              "Infection.Status",  # x-variable
                              # "Diet",  # y-variable
                              facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                              facet.var.y = "Sex",  # Optional
                              # colors = "Sex",  # Optional, col.diet or "Sex"
                              # title = "",  # Optional
                              title = title,
                              # title = NULL,  # Optional.
                              caption = "0 = uninfected, 1 = infected"  # Optional
                              )
## for reproducibility
# p1 <- grouped_ggbarstats(
#   data         = df.expFin,
#   x            = Infection.Status,
#   y            = Sex,
#   type = "np",
#   grouping.var = Diet,
#   package      = "RColorBrewer",
#   palette      = "Dark2"
# )

# mod %>% report()

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# p1


# ggcoefstats(mod) 


# Stats Table
summary(mod) 
Anova(mod, type = 2) 

# disp_stats(p1)

```

\newpage

### Does sex affect infection status?
* infection.status ~ diet

```{r Infection-Sex, message=FALSE, warning=FALSE, include=FALSE}
# GLM
# mod <- gen_glm(data = dt.expFin,
#               x.var = "Diet", # singular or multiple variables as a list. Must be NULL if loop var is the only X-var. Default is NULL
#               y.var = "factor(Infection.Status)",
#               # loop = "x",  # Which variable to loop? "x" or "y", NA is default
#               # loop.var = methods.alpha, # List of variables to loop on
#               # extra.data = NA,  # need to merge dataframes on a sample or ID name? Enter datatable
#               # samp.name="Sample"  # Which columns do you want to merge on? Need to be the same (eg. "Sample")
#               terms = 1,  # How many interaction terms are you using? Default is 1
#               glm.fam = "binomial",  # What GLM family? Default is "gaussian"
#               # rand.eff = NA  # Random effect? Enter which variable you want to be a random effect
#               # fact.y = T
#               run.anv = F
#               )

data <- subset(dt.expFin)
formula = "Infection.Status ~ Sex"
title = "Infection.Status ~ Sex"

mod <- glm( factor(Infection.Status) ~ Sex,
            data = data,
            family = binomial(link="logit"))

# Plot
plot <- gen_bar_plot(data = data,
                              "Infection.Status",  # x-variable
                              # "Diet",  # y-variable
                              facet.var.x = c("Sex"),  # Optional: c("Diet", "Sex") or "Diet"
                              # facet.var.y = "Sex",  # Optional
                              colors = col.sex,  # Optional, col.diet or "Sex"
                              # title = "",  # Optional
                              title = title,
                              # title = NULL,  # Optional.
                              caption = "0 = uninfected, 1 = infected"  # Optional
                              )
## for reproducibility
# p1 <- grouped_ggbarstats(
#   data         = df.expFin,
#   x            = Infection.Status,
#   y            = Sex,
#   type = "np",
#   grouping.var = Diet,
#   package      = "RColorBrewer",
#   palette      = "Dark2"
# )

# mod %>% report()

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# p1


# ggcoefstats(mod) 


# Stats Table
summary(mod)
Anova(mod, type = 2)

# disp_stats(p1)

```


\newpage


### Does infection status associate with physiology?
* Body.Condition.Score ~ infection.status

```{r BodyCond-Infection, message=FALSE, warning=FALSE, include=FALSE}

data = df.expFin
formula = "Body.Condition.Score ~ Infection.Status"

# lm()
mod <- lm( data = data, Body.Condition.Score ~ Infection.Status)


# LM stats table

# Plot
plot <- gen_box_plot_2(data = df.expFin,
                      "Infection.Status",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      # facet.var.x = "Sex",  # Optional
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = formula,  # Optional
                      # title = NULL,  # Optional.
                      caption = "0 = uninfected, 1 = infected"  # Optional
                      )

# mod %>% report()
```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

# Stats Table
summary(mod)
Anova(mod, type = 2)

# ggcoefstats(mod) 
# disp_stats(p1)

```

\newpage

### Does infection status associate with physiology in males?
* Body.Condition.Score ~ infection.status

```{r BodyCond-Infection-males, message=FALSE, warning=FALSE, include=FALSE}

data = subset(df.expFin, Sex == "M")
formula = "Body.Condition.Score ~ Infection.Status"

# lm()
mod <- lm( data = data, Body.Condition.Score ~ Infection.Status )


# LM stats table

# Plot
plot <- gen_box_plot_2(data = data,
                      "Infection.Status",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      # facet.var.x = "Sex",  # Optional
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = formula,  # Optional
                      # title = NULL,  # Optional.
                      caption = "Only males"  # Optional
                      )

# mod %>% report()
```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

# Stats Table
summary(mod)
Anova(mod, type = 2)

# ggcoefstats(mod) 
# disp_stats(p1)

```

\newpage

### Does infection status associate with physiology in males depending on diet?
* Body.Condition.Score ~ infection.status

```{r BodyCond-Infection-Diet-males, message=FALSE, warning=FALSE, include=FALSE}

data = subset(df.expFin, Sex == "M")
formula = "Body.Condition.Score ~ Infection.Status*Diet"
title = "Body.Condition.Score ~ Infection.Status*Diet"

# lm()
mod <- lm( data = data, Body.Condition.Score ~ Infection.Status*Diet )


# LM stats table

# Plot
plot <- gen_box_plot_2(data = data,
                      "Infection.Status",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = "Diet",  # Optional
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = title,  # Optional
                      # title = NULL,  # Optional.
                      caption = "Only males"  # Optional
                      ) + theme(legend.position = "none")

# mod %>% report()
```

```{r echo=FALSE, fig.height=5, fig.width=8}
# Plot
plot

# Stats Table
summary(mod)
Anova(mod, type = 2)

# ggcoefstats(mod) 
# disp_stats(p1)

```

\newpage



### Does diet affect severity score depending on sex?
* severity.score ~ diet

```{r Severity-Diet, message=FALSE, warning=FALSE, include=FALSE}
# Tom's Code:
# # following
# # https://stats.idre.ucla.edu/r/dae/ordinal-logistic-regression/
# mod <- polr( factor(Myco.Severity.Score) ~ Diet*Myco.Dose, data = data, Hess=TRUE)
# pander(summary(mod))
# ctable <- coef(summary(mod))
# p <- pnorm(abs(ctable[,"t value"]), lower.tail=FALSE) *2
# pander(ctable <- cbind(ctable, "p value" = p))

data <- dt.expFin
title = "Severity.Score ~ Diet*Sex"

mod <- polr( factor(Severity.Score) ~ Diet*Sex, data = data, Hess=TRUE)

# Plot
plot <- gen_bar_plot(data = df.expFin,
                              "factor(Severity.Score)",  # x-variable
                              # "Diet",  # y-variable
                              facet.var.x = c("Diet"),  # Optional: c("Diet", "Sex") or "Diet"
                              facet.var.y = "Sex",  # Optional
                              colors = col.severity,  # Optional, col.diet or "Sex"
                              # title = "",  # Optional
                              title = title,  # Optional.
                              caption = "polr()"  # Optional
                              )



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

# Stats Table
summary(mod) 
Anova(mod, type = 2)

```



\newpage

### Does Exposure status associate with physiology in males depending on diet?
* Body.Condition.Score ~ infection.status

```{r BodyCond-Exposure-Diet-males, message=FALSE, warning=FALSE, include=FALSE}

data = subset(df.T1, Sex == "M")
formula = "Body.Condition.Score ~ Exposure*Diet"
title = "Body.Condition.Score ~ Exposure*Diet"

# lm()
mod <- lm( data = data, Body.Condition.Score ~ Exposure*Diet )


# LM stats table

# Plot
plot <- gen_box_plot_2(data = data,
                      "Exposure",  # x-variable
                      "Body.Condition.Score",  # y-variable
                      facet.var.x = "Diet",  # Optional
                      # facet.var.y = "Sex",  # Optional
                      # colors = "Sex",  # Optional, col.diet or "Sex"
                      title = title  # Optional
                      # title = NULL,  # Optional.
                      # caption = "0 = uninfected, 1 = infected. Only males"  # Optional
                      )

# mod %>% report()
```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

# Stats Table
summary(mod)
Anova(mod, type = 2)

# ggcoefstats(mod) 
# disp_stats(p1)

```

\newpage

# Is the microbiome intertwined with these diet by end point associations?

## Does diet impact gut microbiome diversity/composition?
* alpha.diversity ~ diet

### Does diet affect microbiome diversity?


```{r Alpha-Diet, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Diet",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})



plot <- list()
plot <- lapply(names(methods.alpha), function(x){
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Diet",  # x-variable
                              "Alpha.Score",  # y-variable
                              # facet.var.x = "Sex",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.diet,  # Optional
                              title = "glm(Alpha.Score ~ Diet)",  # Optional
                              # title = NULL  # Optional,
                              y.lab = "Diversity (Shannon Entropy)",
                              comparisons = list( c("Gemma", "Watts"), 
                                                  c("Watts", "ZIRC"), 
                                                  c("Gemma", "ZIRC") )
                              ) #+  ggpubr::stat_compare_means(comparisons = comparisons, label = "p.signif", inherit.aes = T)
}) 

# plot <- gen_box_plot_2(data = data,
#                               "Diet",  # x-variable
#                               "Alpha.Score",  # y-variable
#                               # facet.var.x = "Sex",  # Optional
#                               facet.var.y = "Alpha.Metric",  # Optional
#                               colors = col.diet,  # Optional
#                               title = "glm(Alpha.Score ~ Diet)"  # Optional
#                               # title = NULL  # Optional
#                               )



```

```{r echo=FALSE, fig.height=5, fig.width=5}

# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```



\newpage

### Does diet affect microbiome diversity (initial, tank effects)? 

```{r Alpha-Diet-initial-randEff, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.T0.melt

mod.list <- lapply(methods.alpha, function(alpha){
  lme4::glmer( formula = "Alpha.Score ~ Diet + (1 | Tank)",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})






# Build Plot
plot <- ggplot(data = dt.alphaPlus.conT0T1.melt,
              aes_string(x = "Diet", y = "Alpha.Score")) +
    geom_boxplot(aes(fill = Tank ) # converts string to var name
    ) +
    # geom_smooth(aes(color = Tank),
    #             method="glm", 
    #             size = 1, 
    #             alpha = .20) + 
    facet_grid(Alpha.Metric ~ ., scales = "free_y") + 
    scale_fill_manual(values = c(pal.dark2, pal.set1)) +
    # scale_color_manual(values = c(pal.paired)) +
    guides(color=F) +
    # theme(legend.position = "non") +
    labs(
      title = "glm(Alpha.Score ~ Diet + (1 | Tank) )",
      y = "Alpha Score",
      caption = "Initial time point"
    )

```




```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})





```

\newpage

### Does diet affect microbiome diversity (final, tank effects)? 

```{r Alpha-Diet-final-randEff, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  lme4::glmer( formula = "Alpha.Score ~ Diet + (1 | Tank)",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})

# Must place this code after linear model
if(isTRUE(plot.oneAlpha)){
  data <- subset(data, Alpha.Metric == subset.alphaMetric)
}
####


# Build Plot
plot <- ggplot(data = dt.alphaPlus.conT0T1.melt,
              aes_string(x = "Diet", y = "Alpha.Score")) +
    geom_boxplot(aes(fill = Tank ) # converts string to var name
    ) +
    # geom_smooth(aes(color = Tank),
    #             method="glm", 
    #             size = 1, 
    #             alpha = .20) + 
    facet_grid(Alpha.Metric ~ ., scales = "free_y") + 
    scale_fill_manual(values = c(pal.dark2, pal.set1)) +
    # scale_color_manual(values = c(pal.paired)) +
    guides(color=F) +
    # theme(legend.position = "non") +
    labs(
      title = "glm(Alpha.Score ~ Diet + (1 | Tank) )",
      y = "Alpha Score",
      caption = "Final time point"
    )

```




```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})





```

\newpage

### Does time affect microbiome diversity?


```{r Alpha-time, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.conT0T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Timepoint",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})

plot <- list()
plot <- lapply(names(methods.alpha), function(x){
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Timepoint",  # x-variable
                              "Alpha.Score",  # y-variable
                              # facet.var.x = "Sex",  # Optional
                              facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.timepoint,  # Optional
                              title = "glm(Alpha.Score ~ Timepoint)"  # Optional
                              # title = NULL  # Optional
                              )
})

# plot <- gen_box_plot_2(data = data,
#                               "Timepoint",  # x-variable
#                               "Alpha.Score",  # y-variable
#                               # facet.var.x = "Sex",  # Optional
#                               facet.var.y = "Alpha.Metric",  # Optional
#                               colors = col.timepoint,  # Optional
#                               title = "glm(Alpha.Score ~ Timepoint)"  # Optional
#                               # title = NULL  # Optional
#                               )



```


```{r echo=FALSE, fig.height=5, fig.width=5}

# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

\newpage

### Does diet affect microbiome diversity depending on time point?
```{r Alpha-Diet-time, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.conT0T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Diet*Timepoint",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})




plot <- list()
plot <- lapply(names(methods.alpha), function(x){
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Diet",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Timepoint",  # Optional
                              facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.diet,  # Optional
                              title = "glm(Alpha.Score ~ Diet*Timepoint)",  # Optional
                              # title = NULL  # Optional,
                              comparisons = list( c("3mpf", "6mpf")#, 
                                                  # c("Watts", "ZIRC"), 
                                                  # c("Gemma", "ZIRC") 
                                                  )
                              )
})

plot2 <- list()
plot2 <- lapply(names(methods.alpha), function(x){
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Timepoint",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Diet",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.timepoint,  # Optional
                              title = "glm(Alpha.Score ~ Timepoint*Diet)",  # Optional
                              # title = NULL  # Optional
                              y.lab = "Diversity (Shannon Entropy)",
                              comparisons = list( c("3mpf", "6mpf")#, 
                                                  # c("Watts", "ZIRC"), 
                                                  # c("Gemma", "ZIRC") 
                                                  )
                              ) #+ scale_x_discrete(labels = c("3mpf", "6mpf"))
})

# plot <- gen_box_plot_2(data = data,
#                               "Diet",  # x-variable
#                               "Alpha.Score",  # y-variable
#                               facet.var.x = "Timepoint",  # Optional
#                               facet.var.y = "Alpha.Metric",  # Optional
#                               colors = col.diet,  # Optional
#                               title = "glm(Alpha.Score ~ Diet*Timepoint)"  # Optional
#                               # title = NULL  # Optional
#                               )


# plot2 <- gen_box_plot_2(data = data,
#                               "Timepoint",  # x-variable
#                               "Alpha.Score",  # y-variable
#                               facet.var.x = "Diet",  # Optional
#                               facet.var.y = "Alpha.Metric",  # Optional
#                               colors = col.timepoint,  # Optional
#                               title = "glm(Alpha.Score ~ Timepoint*Diet)"  # Optional
#                               # title = NULL  # Optional
#                               )


```

```{r echo=FALSE, fig.height=5, fig.width=8}

# Plot
plot
plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

\newpage


### Does sex affect microbiome diversity?


```{r Alpha-Sex, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.T0.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Sex",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})

plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Sex",  # x-variable
                              "Alpha.Score",  # y-variable
                              # facet.var.x = "Sex",  # Optional
                              facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.sex,  # Optional
                              title = "glm(Alpha.Score ~ Sex)"  # Optional
                              # title = NULL  # Optional
                              )
})

# plot2 <- list()
# plot2 <- lapply(names(methods.alpha), function(x){
#   gen_box_plot_2(data = data,
#                               "Timepoint",  # x-variable
#                               "Alpha.Score",  # y-variable
#                               facet.var.x = "Diet",  # Optional
#                               facet.var.y = "Alpha.Metric",  # Optional
#                               colors = col.timepoint,  # Optional
#                               title = "glm(Alpha.Score ~ Timepoint*Diet)"  # Optional
#                               # title = NULL  # Optional
#                               )
# })
# 
# plot <- gen_box_plot_2(data = data,
#                               "Sex",  # x-variable
#                               "Alpha.Score",  # y-variable
#                               # facet.var.x = "Sex",  # Optional
#                               facet.var.y = "Alpha.Metric",  # Optional
#                               colors = col.sex,  # Optional
#                               title = "glm(Alpha.Score ~ Sex)"  # Optional
#                               # title = NULL  # Optional
#                               )


```


```{r echo=FALSE, fig.height=5, fig.width=5}

# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

\newpage

### Does diet affect microbiome diversity depending on Sex?
```{r Alpha-Diet-Sex, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.T0.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Diet*Sex",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})


  
plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
   gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Diet",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Sex",  # Optional
                              facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.diet,  # Optional
                              title = "glm(Alpha.Score ~ Diet*Sex)"  # Optional
                              # title = NULL  # Optional
                              )
})

plot2 <- list()
plot2 <- lapply(names(methods.alpha), function(x){
  # subset(data, Alpha.Metric == x)
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Sex",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Diet",  # Optional
                              facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.sex,  # Optional
                              title = "glm(Alpha.Score ~ Sex*Diet)"  # Optional
                              # title = NULL  # Optional
                              )
})


# plot <- gen_box_plot_2(data = data,
#                               "Diet",  # x-variable
#                               "Alpha.Score",  # y-variable
#                               facet.var.x = "Sex",  # Optional
#                               facet.var.y = "Alpha.Metric",  # Optional
#                               colors = col.diet,  # Optional
#                               title = "glm(Alpha.Score ~ Diet*Sex)"  # Optional
#                               # title = NULL  # Optional
#                               )


# plot2 <- gen_box_plot_2(data = data,
#                               "Sex",  # x-variable
#                               "Alpha.Score",  # y-variable
#                               facet.var.x = "Diet",  # Optional
#                               facet.var.y = "Alpha.Metric",  # Optional
#                               colors = col.sex,  # Optional
#                               title = "glm(Alpha.Score ~ Sex*Diet)"  # Optional
#                               # title = NULL  # Optional
#                               )


```

```{r echo=FALSE, fig.height=5, fig.width=5}

# Plot
plot
plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

\newpage

### Does diet affect microbiome composition (initial time point)?
* beta.diversity ~ diet

```{r Beta-Diet-initial, message=FALSE, warning=FALSE, include=FALSE}

data <- df.T0
ps.obj <- ps.T0
dist <- distList.T0

# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 1,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)



# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Diet",
                            # x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.diet
                            )

plot2 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "Diet",
                        # shape = "Exposure", 
                        title = "",
                        colors = col.diet
                        )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2

# Stats Table
stats_table_3(anova)

anova %>% pander()

beta_hom_disper(data = data,
                physeq = ps.T0,
                vars = "Diet",
                methods = c("bray",
                            "canberra",
                            "sor"
                            )) 

```

\newpage

### Does diet affect microbiome composition (final time point)?
* beta.diversity ~ diet

```{r Beta-Diet-final, message=FALSE, warning=FALSE, include=FALSE}

data <- df.T1
ps.obj <- ps.T1
dist <- distList.T1

# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 1,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Diet",
                            # x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.diet
                            )

plot2 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "Diet",
                        # shape = "Exposure", 
                        title = "",
                        colors = col.diet
                        )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2

# Stats Table
stats_table_3(anova)

anova %>% pander()

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = "Diet",
                methods = c("bray",
                            "canberra",
                            "sor"
                            )) 

```

\newpage



### Does time affect microbiome composition?


```{r Beta-Time, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
ps.obj <- ps.conT0T1
dist <- distList.conT0T1


# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Timepoint"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Timepoint",
                            # x.var.2 = "Timepoint",  # Optional
                            y.var = methods.beta,
                            elps.var = "Timepoint",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            title = "Beta.Score ~ Timepoint",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.timepoint
                            )

# plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
#                             x.var = "Timepoint",
#                             # x.var.2 = "Diet",  # Optional
#                             y.var = methods.beta,
#                             elps.var = "Diet",  # Optional
#                             axis = beta.axis.full,
#                             stats = anova,  # Optional
#                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional
#                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
#                             disp.stats = T,  # Optional
#                             colors = col.timepoint
#                             )

plot2 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Timepoint",
                        # x.var.2 = "Timepoint",
                        title = "Beta.Score ~ Timepoint",
                        colors = col.timepoint
                        )




```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# plot3
# plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Diet"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Timepoint"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)


```


\newpage


### Does sex affect microbiome composition?


```{r Beta-Sex, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
ps.obj <- ps.conT0T1
dist <- distList.conT0T1


# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Sex"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Sex",
                            # x.var.2 = "Timepoint",  # Optional
                            y.var = methods.beta,
                            elps.var = "Sex",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            title = "Beta.Score ~ Sex",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.sex
                            )

# plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
#                             x.var = "Timepoint",
#                             # x.var.2 = "Diet",  # Optional
#                             y.var = methods.beta,
#                             elps.var = "Diet",  # Optional
#                             axis = beta.axis.full,
#                             stats = anova,  # Optional
#                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional
#                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
#                             disp.stats = T,  # Optional
#                             colors = col.timepoint
#                             )

plot2 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Sex",
                        # x.var.2 = "Timepoint",
                        title = "Beta.Score ~ Sex",
                        colors = col.sex
                        )




```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# plot3
# plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Diet"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Timepoint"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)


```

\newpage

### Does diet affect microbiome composition depending on sex?
* beta.diversity ~ diet*time

```{r Beta-Diet-sex, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
ps.obj <- ps.conT0T1
dist <- distList.conT0T1


# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Diet", "Sex"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Diet",
                            x.var.2 = "Sex",  # Optional
                            y.var = methods.beta,
                            elps.var = "Sex",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.diet
                            )

plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Sex",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.sex
                            )

plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "Sex",
                        title = "",
                        colors = col.diet
                        )

plot4 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Sex",
                        x.var.2 = "Diet",
                        title = "",
                        colors = col.sex
                        )



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
plot3
plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Diet"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Sex"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)


```


\newpage


### Does diet affect microbiome composition depending on time?
* beta.diversity ~ diet*time

```{r Beta-Diet-Time, message=FALSE, warning=FALSE, include=FALSE}

data <- df.conT0T1
ps.obj <- ps.conT0T1
dist <- distList.conT0T1


# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Diet", "Timepoint"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Diet",
                            x.var.2 = "Timepoint",  # Optional
                            y.var = methods.beta,
                            elps.var = "Timepoint",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.diet
                            )

plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Timepoint",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.timepoint
                            )

plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "Timepoint",
                        title = "",
                        colors = col.diet
                        )

plot4 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Timepoint",
                        x.var.2 = "Diet",
                        title = "",
                        colors = col.timepoint
                        )



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
plot3
plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Diet"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Timepoint"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)


```


\newpage

### Does diet affect microbiome differential abundance (initial and final)?

```{r diffAbund-diet, echo=FALSE, fig.height=10, fig.width=6, message=FALSE, warning=FALSE}
plot_diffAbund_2(DiffAbund_Diet[["Sig.Res"]] )
plot_diffAbund_2(DiffAbund_Diet_rEff[["Sig.Res"]] )

```


```{r eval=FALSE, include=FALSE}
plot_diffAbund_2(DiffAbund_Diet_rEff[["Sig.Res"]])
```



```{r}
# DiffAbund_Diet_rEff[["Stats.Table"]] 
```





\newpage

### Does diet affect microbiome differential abundance depending on time (initial)?

```{r diffAbund-diet-time-init, echo=FALSE, message=FALSE, warning=FALSE}

plot_diffAbund_2(DiffAbund_DietInitial[["Sig.Res"]] )
plot_diffAbund_2(DiffAbund_DietInitial_rEff[["Sig.Res"]] )

```

```{r echo=FALSE}
# DiffAbund_DietInitial_rEff[["Stats.Table"]]
```


\newpage

### Does diet affect microbiome differential abundance depending on time (final)?

```{r diffAbund-diet-time-final, echo=FALSE, fig.height=10, fig.width=6, message=FALSE, warning=FALSE}

plot_diffAbund_2(DiffAbund_DietFinal[["Sig.Res"]] )
plot_diffAbund_2(DiffAbund_DietFinal_rEff[["Sig.Res"]] )
# DiffAbund_DietFinal[["Heatmap"]]




```


```{r}
# DiffAbund_DietFinal_rEff[["Stats.Table"]]
```



\newpage

### Does diet affect microbiome differential abundance depending on time (initial and final)?

```{r diffAbund-diet-time-2, echo=FALSE, fig.height=10, fig.width=6, message=FALSE, warning=FALSE}

plot_diffAbund_2(DiffAbund_DietTime[["Sig.Res"]] )
plot_diffAbund_2(DiffAbund_DietTime_rEff[["Sig.Res"]] )

# DiffAbund_DietTime[["Heatmap"]]



```


```{r}
# DiffAbund_DietTime_rEff[["Stats.Table"]]
```



\newpage

### Does diet affect microbiome differential abundance depending on sex?


```{r diffAbund-diet-sex, echo=FALSE, message=FALSE, warning=FALSE}

plot_diffAbund_2(DiffAbund_Sex[["Sig.Res"]] )
# plot_diffAbund_2(DiffAbund_Sex_rEff[["Sig.Res"]] )  # No sig effects with random effects

# DiffAbund_DietTime[["Heatmap"]]



```


```{r}
# DiffAbund_Sex[["Stats.Table"]]
```
\newpage


## Does variation in the microbiome link to variation in body condition score?
* Body.Condition.Score ~ alpha.diversity
  * Does it depend on diet?



### Body condition vs alpha diversity

```{r Alpha-BodyCond, message=FALSE, warning=FALSE, include=FALSE}


data <- dt.alphaPlus.conT0T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Body.Condition.Score ~ Alpha.Score",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})

plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
   gen_scatter_plot_2(data = subset(data, Alpha.Metric == x),
                          "Alpha.Score",  # x-variable
                          "Body.Condition.Score",  # y-variable
                          # facet.var.x = "Sex",  # Optional
                          facet.var.y = "Alpha.Metric",  # Optional
                          colors = col.diverge,  # Optional
                          title = "glm(Body.Condition.Score ~ Alpha.Score)"  # Optional
                          # title = NULL  # Optional
                          )  #+ theme(legend.position = "bottom")
})







```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# p1

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```


```{r eval=FALSE, include=FALSE}
lapply(methods.alpha, function(x){
  mod.list[[x]] %>% report()
})
```



\newpage



### Body condition vs alpha diversity depending on timepoint 
(color = time)


```{r Alpha-BodyCond-Diet-time, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.conT0T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Body.Condition.Score ~ Alpha.Score*Timepoint",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})

plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
   gen_scatter_plot_2(data = subset(data, Alpha.Metric == x),
                          "Alpha.Score",  # x-variable
                          "Body.Condition.Score",  # y-variable
                          # facet.var.x = "Sex",  # Optional
                          facet.var.y = "Alpha.Metric",  # Optional
                          colors = col.diverge,  # Optional
                          title = "glm(Body.Condition.Score ~ Alpha.Score)"  # Optional
                          # title = NULL  # Optional
                          )  #+ theme(legend.position = "bottom")
})

plot2 <- list()
plot2 <- lapply(names(methods.alpha), function(x){
  # subset(data, Alpha.Metric == x)
  gen_scatter_plot_2(data = subset(data, Alpha.Metric == x),
                          "Alpha.Score",  # x-variable
                          "Body.Condition.Score",  # y-variable
                          facet.var.x = "Timepoint",  # Optional
                          facet.var.y = "Alpha.Metric",  # Optional
                          colors = col.diet,  # Optional
                          title = "glm(Body.Condition.Score ~ Alpha.Score*Timepoint)"  # Optional
                          # title = NULL  # Optional
                          )  #+ theme(legend.position = "bottom")
})



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

```{r eval=FALSE, include=FALSE}
lapply(methods.alpha, function(x){
  mod.list[[x]] %>% report()
})
```

\newpage

### Body condition vs alpha diversity depending on sex (control)



```{r Alpha-BodyCond-sex, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.T0.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Body.Condition.Score ~ Alpha.Score*Sex",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})

plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
   gen_scatter_plot_2(data = subset(data, Alpha.Metric == x),
                          "Alpha.Score",  # x-variable
                          "Body.Condition.Score",  # y-variable
                          facet.var.x = "Sex",  # Optional
                          facet.var.y = "Alpha.Metric",  # Optional
                          colors = col.sex,  # Optional
                          title = "glm(Body.Condition.Score ~ Alpha.Score*Sex)"  # Optional
                          # title = NULL  # Optional
                          )  #+ theme(legend.position = "bottom")
})



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

```{r eval=FALSE, include=FALSE}
lapply(methods.alpha, function(x){
  mod.list[[x]] %>% report()
})
```

\newpage

### Body condition vs alpha diversity depending on sex and diet



```{r Alpha-BodyCond-diet-sex, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.conT0T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Body.Condition.Score ~ Alpha.Score*Sex*Diet",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})



plot <- ggplot(data = data, aes_string(x = "Alpha.Score", y = "Body.Condition.Score")) +
    geom_point(aes(color = Sex ) # converts string to var name
    ) +
    geom_smooth(aes(color = Sex),
                method="glm",
                size = 1,
                alpha = .20) +
    facet_grid(Alpha.Metric ~ Diet, scales = "free_y") +
    scale_fill_brewer(palette = "Dark2") +
    scale_color_brewer(palette = "Dark2") +
    guides(fill=guide_legend(title="Alpha.Score")) +
    # theme(legend.position = "non") +
    labs(
      title = "glm(Body.Condition.Score ~ Alpha.Score*Sex*Diet)",
      # title = ifelse(title != "", title, ""),
      y = "Body.Condition.Score"
    )



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

```{r eval=FALSE, include=FALSE}
lapply(methods.alpha, function(x){
  mod.list[[x]] %>% report()
})
```

\newpage

### Body condition vs alpha diversity depending on diet 



```{r Alpha-BodyCond-Diet, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.conT0T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Body.Condition.Score ~ Alpha.Score*Diet",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})

plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
   gen_scatter_plot_2(data = subset(data, Alpha.Metric == x),
                          "Alpha.Score",  # x-variable
                          "Body.Condition.Score",  # y-variable
                          facet.var.x = "Diet",  # Optional
                          facet.var.y = "Alpha.Metric",  # Optional
                          colors = col.diet,  # Optional
                          title = "glm(Body.Condition.Score ~ Alpha.Score*Diet)"  # Optional
                          # title = NULL  # Optional
                          )  #+ theme(legend.position = "bottom")
})


```





```{r echo=FALSE, fig.height=5, fig.width=8}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))



```

```{r eval=FALSE, include=FALSE}
lapply(methods.alpha, function(x){
  mod.list[[x]] %>% report()
})
```

\newpage

### Body condition vs alpha diversity in control ZIRC fed fish



```{r Alpha-BodyCond-ZIRC, message=FALSE, warning=FALSE, include=FALSE}

data <- subset(dt.alphaPlus.conT0T1.melt, Diet == "ZIRC")
# data <- subset(dt.alphaPlus.T0.melt, Diet == "ZIRC")

# kruskal.test(Body.Condition.Score ~ Timepoint, data = subset(dt.alphaPlus.conT0T1.melt, Diet == "ZIRC"))

# summary(data$Body.Condition.Score)
# BC.min <- min(data$Body.Condition.Score)
# BC.1 <- quantile(data$Body.Condition.Score, 0.25)
BC.med <- median(data$Body.Condition.Score)
# BC.3 <- quantile(data$Body.Condition.Score, 0.75)
# BC.max <- max(data$Body.Condition.Score)

# data <- data %>%
#     mutate(Body.Cond_lvl = case_when(
#     (Body.Condition.Score >= BC.min & Body.Condition.Score < BC.1) ~ "1",
#     (Body.Condition.Score >= BC.1 & Body.Condition.Score < BC.med) ~ "2",
#     (Body.Condition.Score >= BC.med & Body.Condition.Score < BC.3) ~ "3",
#     (Body.Condition.Score >= BC.3 & Body.Condition.Score <= BC.max) ~ "4"),
#     .after = Body.Condition.Score)

data <- data %>%
    mutate(Body.Cond_lvl = case_when(
    Body.Condition.Score < BC.med ~ "< median",
    Body.Condition.Score >= BC.med ~ ">= median"),
    .after = Body.Condition.Score)



mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Body.Cond_lvl*Sex",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})



plot <- list()
plot <- lapply(names(methods.alpha), function(x){
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Body.Cond_lvl",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Sex",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = c(col.diverge[c(3,7)]),  # Optional
                              # title = "glm(Alpha.Score ~ Diet)",  # Optional
                              # title = NULL  # Optional,
                              y.lab = "Diversity (Shannon Entropy)",
                              caption = paste0("Median Body Cond = ", round(BC.med,3)),
                              comparisons = list( c("< median", ">= median")#, 
                                                  # c("Watts", "ZIRC"), 
                                                  # c("Gemma", "ZIRC") 
                                                  )
                              ) + labs(x = "Body Condition Levels")
}) 

plot2 <- list()
plot2 <- lapply(names(methods.alpha), function(x){
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Sex",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Body.Cond_lvl",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.timepoint,  # Optional
                              # title = "glm(Alpha.Score ~ Diet)",  # Optional
                              # title = NULL  # Optional,
                              y.lab = "Diversity (Shannon Entropy)",
                              caption = paste0("Median Body Cond = ", round(BC.med,3)),
                              comparisons = list( c("M", "F")#, 
                                                  # c("Watts", "ZIRC"), 
                                                  # c("Gemma", "ZIRC") 
                                                  )
                              ) + labs(x = "Body Condition Levels")
}) 

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})



stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))


```

\newpage

### Body condition vs alpha diversity in control ZIRC fed fish, sex



```{r Alpha-BodyCond-ZIRC-sex, message=FALSE, warning=FALSE, include=FALSE}

data <- subset(dt.alphaPlus.conT0T1.melt, Diet == "ZIRC")
# data <- subset(dt.alphaPlus.T0.melt, Diet == "ZIRC")

# kruskal.test(Body.Condition.Score ~ Timepoint, data = subset(dt.alphaPlus.conT0T1.melt, Diet == "ZIRC"))

# summary(data$Body.Condition.Score)
# BC.min <- min(data$Body.Condition.Score)
# BC.1 <- quantile(data$Body.Condition.Score, 0.25)
BC.med <- median(data$Body.Condition.Score)
# BC.3 <- quantile(data$Body.Condition.Score, 0.75)
# BC.max <- max(data$Body.Condition.Score)

# data <- data %>%
#     mutate(Body.Cond_lvl = case_when(
#     (Body.Condition.Score >= BC.min & Body.Condition.Score < BC.1) ~ "1",
#     (Body.Condition.Score >= BC.1 & Body.Condition.Score < BC.med) ~ "2",
#     (Body.Condition.Score >= BC.med & Body.Condition.Score < BC.3) ~ "3",
#     (Body.Condition.Score >= BC.3 & Body.Condition.Score <= BC.max) ~ "4"),
#     .after = Body.Condition.Score)

# data <- data %>%
#     mutate(Body.Cond_lvl = case_when(
#     Body.Condition.Score < BC.med ~ "< median",
#     Body.Condition.Score >= BC.med ~ ">= median"),
#     .after = Body.Condition.Score)



mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Body.Condition.Score*Sex",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})



plot <- list()
plot <- lapply(names(methods.alpha), function(x){
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Body.Condition.Score",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Sex",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = c(col.diverge[c(3,7)]),  # Optional
                              # title = "glm(Alpha.Score ~ Diet)",  # Optional
                              # title = NULL  # Optional,
                              y.lab = "Diversity (Shannon Entropy)",
                              caption = paste0("Median Body Cond = ", round(BC.med,3)),
                              comparisons = list( c("M", "F")#, 
                                                  # c("Watts", "ZIRC"), 
                                                  # c("Gemma", "ZIRC") 
                                                  )
                              ) + labs(x = "Body Condition Levels")
}) 

plot2 <- list()
plot2 <- lapply(names(methods.alpha), function(x){
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Sex",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Body.Condition.Score",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.timepoint,  # Optional
                              # title = "glm(Alpha.Score ~ Diet)",  # Optional
                              # title = NULL  # Optional,
                              y.lab = "Diversity (Shannon Entropy)",
                              caption = paste0("Median Body Cond = ", round(BC.med,3)),
                              comparisons = list( c("M", "F")#, 
                                                  # c("Watts", "ZIRC"), 
                                                  # c("Gemma", "ZIRC") 
                                                  )
                              ) + labs(x = "Body Condition Levels")
}) 

```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})



stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))


```


\newpage

```{r fig.height=5, fig.width=5}
# Build Plot
# plot <- gen_scatter_plot_2(data = data,
#                           "Alpha.Score",  # x-variable
#                           "Body.Condition.Score",  # y-variable
#                           # facet.var.x = "Body.Cond_lvl",  # Optional
#                           facet.var.y = "Timepoint",  # Optional
#                           # colors = col.diet,  # Optional
#                           title = "glm(Body.Condition.Score ~ Alpha.Score)",  # Optional
#                           # title = NULL  # Optional
#                           # y.lab = "Diversity (Shannon Entropy)"
#                           )  #+ theme(legend.position = "bottom")


ggplot(data = subset(data, Alpha.Metric == "Shannon"),
              aes_string(x = "Alpha.Score", y = "Body.Condition.Score")) +
    geom_point(aes(color = Timepoint ) # converts string to var name
    ) +
    geom_smooth(aes(color = Timepoint),
                method="glm",
                size = 1,
                alpha = .20) +
    facet_grid(Alpha.Metric ~ ., scales = "free_y") + 
    scale_color_manual(values = c(col.timepoint), name = "Timepoint", labels = c("3mpf", "6mpf")) +
    # scale_color_manual(values = c(pal.paired)) +
    theme(legend.position = "bottom") +
    labs(
      #title = "glm(Alpha.Score ~ Diet + (1 | Tank) )",
      x = "Diversity (Shannon Entropy)",
      #caption = "Initial time point"
    )

ggplot(data = subset(data, Alpha.Metric == "Shannon"),
              aes_string(x = "Alpha.Score", y = "Body.Condition.Score")) +
    geom_point(aes(color = Timepoint ) # converts string to var name
    ) +
    geom_smooth(aes(color = Timepoint),
                method="glm",
                size = 1,
                alpha = .20) +
    facet_grid(Alpha.Metric ~ ., scales = "free_y") + 
    scale_color_manual(values = c(col.timepoint), name = "Timepoint", labels = c("3mpf", "6mpf")) +
    # scale_color_manual(values = c(pal.paired)) +
    theme(legend.position = "bottom") +
    labs(
      #title = "glm(Alpha.Score ~ Diet + (1 | Tank) )",
      x = "Diversity (Shannon Entropy)",
      #caption = "Initial time point"
    )
```



\newpage

### Body condition vs beta diversity in control ZIRC fed fish

```{r Beta-BodyCond-ZIRC, message=FALSE, warning=FALSE, include=FALSE}

data = subset(df.conT0T1, Diet == "ZIRC")
ps.obj = subset_samples(ps.conT0T1, Diet == "ZIRC") 
dist = distList.conT0T1.ZIRC

# Full dbRDA models

data <- data %>%
    mutate(Body.Cond_lvl = case_when(
    Body.Condition.Score < BC.med ~ "< median",
    Body.Condition.Score >= BC.med ~ ">= median"),
    .after = Body.Condition.Score)

sample_data(ps.obj) <- sample.data.frame(ps.obj)  %>%
    mutate(Body.Cond_lvl = case_when(
    Body.Condition.Score < BC.med ~ "< median",
    Body.Condition.Score >= BC.med ~ ">= median"),
    .after = Body.Condition.Score)

# ZIRC only
distList.conT0T1.ZIRC <- gen.dist.matrices(ps.obj, methods = "taxonomic", cores = num.cores)



mod <- full_dbrda_model_2(vars = c("Body.Cond_lvl*Timepoint"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Body.Cond_lvl",
                            x.var.2 = "Timepoint",  # Optional
                            y.var = methods.beta,
                            elps.var = "Timepoint",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = c(col.diverge[c(3,7)])
                            )

plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Timepoint",
                            x.var.2 = "Body.Cond_lvl",  # Optional
                            y.var = methods.beta,
                            elps.var = "Body.Cond_lvl",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T , # Optional
                            colors = col.timepoint
                            )

plot3 <- betaUncon_plot(data = data,
                        # method = "",
                        physeq = ps.obj,
                        # dist = "bray",
                        facet.var.x = ".",
                        facet.var.y = ".",
                        x.var = "Body.Cond_lvl",
                        x.var.2 = "Timepoint",
                        title = "",
                        colors = c(col.diverge[c(3,7)])
                        )

plot4 <- betaUncon_plot(data = data,
                        # method = "",
                        physeq = ps.obj,
                        # dist = "bray",
                        facet.var.x = ".",
                        facet.var.y = ".",
                        x.var = "Timepoint",
                        x.var.2 = "Body.Cond_lvl",
                        title = "",
                        colors = col.timepoint
                        )

```



```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
plot3
plot4

# Stats
stats_table_3(anova)

anova %>% pander()

# How to run numerical variable on beta disp?
beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Body.Cond_lvl"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Timepoint"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

```

\newpage

```{r Beta-BodyCond-ZIRC, message=FALSE, warning=FALSE, include=FALSE}

data <- subset(dt.alphaPlus.conT0T1.melt, Diet == "Zirc")

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Body.Condition.Score ~ Alpha.Score*Sex",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})

# Must place this code after linear model
if(isTRUE(plot.oneAlpha)){
  data <- subset(data, Alpha.Metric == subset.alphaMetric)
}
####

# Build Plot
plot <- gen_scatter_plot_2(data = data,
                          "Alpha.Score",  # x-variable
                          "Body.Condition.Score",  # y-variable
                          facet.var.x = "Sex",  # Optional
                          facet.var.y = "Alpha.Metric",  # Optional
                          colors = col.sex,  # Optional
                          title = "glm(Body.Condition.Score ~ Alpha.Score*Sex)"  # Optional
                          # title = NULL  # Optional
                          )  #+ theme(legend.position = "bottom")



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})





```



\newpage


### Body condition vs alpha diversity depending on diet (Tank effects?) 



```{r Alpha-BodyCond-Diet-randEff, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.conT0T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  lme4::glmer( formula = "Body.Condition.Score ~ Alpha.Score*Diet + (1 | Tank)",
       data = subset(data, Alpha.Metric == alpha),
       family = "gaussian")
})


# Build Plot
plot <- ggplot(data = dt.alphaPlus.conT0T1.melt,
              aes_string(x = "Alpha.Score", y = "Body.Condition.Score")) +
    geom_point(aes(color = Tank ) # converts string to var name
    ) +
    geom_smooth(aes(color = Tank),
                method="glm", 
                size = 1, 
                alpha = .20) + 
    facet_grid(paste0("Alpha.Metric", " ~ ", "Diet"), scales = "free_y") + 
    # scale_fill_manual(values = colors) +
    # scale_color_manual(values = colors) +
    guides(fill=guide_legend(title="Alpha.Score")) +
    # theme(legend.position = "non") +
    labs(
      title = "glm(Body.Condition.Score ~ Alpha.Score*Diet + (1 | Tank))",
      y = "Body.Condition.Score"
    )

```




```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})





```

```{r eval=FALSE, include=FALSE}
lapply(methods.alpha, function(x){
  mod.list[[x]] %>% report()
})
```


```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

# stats_table_3( lapply(methods.alpha, function(x){
#   gen_glm_anova(mod.list[[x]], x)
# }) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

```{r eval=FALSE, include=FALSE}
lapply(methods.alpha, function(x){
  mod.list[[x]] %>% report()
})
```

\newpage

### Does variation in physiology link to variation in microbiome composition?
* Beta.diversity ~ Body.Condition.Score


```{r Beta-BodyCond, message=FALSE, warning=FALSE, include=FALSE}

data = df.conT0T1
ps.obj = ps.conT0T1
dist = distList.conT0T1

# Full dbRDA models

mod <- full_dbrda_model_2(vars = c("Body.Condition.Score"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 1,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Plot
plot <- gen_PCoA_Num_Plot_2(data = beta.dt.full,
                            x.var = "Body.Condition.Score",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )

plot2 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        # x.var.2 = "Timepoint",
                        title = "",
                        colors = col.diet
                        )


```





```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# plot3
# plot4

# Stats
stats_table_3(anova)

anova %>% pander()

# # How to run numerical variable on beta disp?
# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("factor(Body.Condition.Score"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)

```

\newpage

### Does variation in physiology link to variation in microbiome composition and time?
* Beta.diversity ~ Body.Condition.Score


```{r Beta-BodyCond-time, message=FALSE, warning=FALSE, include=FALSE}

data = df.conT0T1
ps.obj = ps.conT0T1
dist = distList.conT0T1

# Full dbRDA models

mod <- full_dbrda_model_2(vars = c("Body.Condition.Score", "Timepoint"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Plot
plot <- gen_PCoA_Num_Plot_2(data = beta.dt.full,
                            x.var = "Body.Condition.Score",
                            x.var.2 = "Timepoint",  # Optional
                            y.var = methods.beta,
                            elps.var = "Timepoint",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )

# plot2 <- gen_PCoA_Num_Plot_2(data = beta.dt.full,
#                             x.var = "Body.Condition.Score",
#                             x.var.2 = "Timepoint",  # Optional
#                             y.var = methods.beta,
#                             elps.var = "Diet",  # Optional
#                             axis = beta.axis.full,
#                             stats = anova,  # Optional
#                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional
#                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
#                             disp.stats = T  # Optional
#                             )


plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Timepoint",
                        # x.var.2 = "Timepoint",
                        title = "",
                        colors = col.diet
                        )

# plot3 <- betaUncon_plot(data = data, 
#                         # method = "", 
#                         physeq = ps.obj, 
#                         # dist = "bray", 
#                         facet.var.x = ".", 
#                         facet.var.y = ".",
#                         x.var = "Body.Condition.Score",
#                         x.var.2 = "Timepoint",
#                         title = "",
#                         colors = col.diet
#                         ) 
```





```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2
plot3
# plot4

# Stats
stats_table_3(anova)

anova %>% pander()


# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Diet"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Timepoint"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)


```

\newpage

### Does variation in physiology link to variation in microbiome composition (Initial)?
* Beta.diversity ~ Body.Condition.Score


```{r Beta-BodyCond-initial, message=FALSE, warning=FALSE, include=FALSE}

data = df.T0
ps.obj = ps.T0
dist = distList.T0

# Full dbRDA models

mod <- full_dbrda_model_2(vars = c("Body.Condition.Score", "Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Plot
plot <- gen_PCoA_Num_Plot_2(data = beta.dt.full,
                            x.var = "Body.Condition.Score",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )

# plot2 <- gen_PCoA_Num_Plot_2(data = beta.dt.full,
#                             x.var = "Body.Condition.Score",
#                             x.var.2 = "Timepoint",  # Optional
#                             y.var = methods.beta,
#                             elps.var = "Diet",  # Optional
#                             axis = beta.axis.full,
#                             stats = anova,  # Optional
#                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional
#                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
#                             disp.stats = T  # Optional
#                             )


plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "",
                        title = "",
                        colors = col.diet
                        )
# plot3

```





```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2
plot3
# plot4

# Stats
stats_table_3(anova)

anova %>% pander()


beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Diet"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

```

\newpage


### Does variation in physiology link to variation in microbiome composition (Final)?
* Beta.diversity ~ Body.Condition.Score


```{r Beta-BodyCond-final, message=FALSE, warning=FALSE, include=FALSE}

data = df.T1
ps.obj = ps.T1
dist = distList.T1

# Full dbRDA models

mod <- full_dbrda_model_2(vars = c("Body.Condition.Score", "Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Plot
plot <- gen_PCoA_Num_Plot_2(data = beta.dt.full,
                            x.var = "Body.Condition.Score",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )

# plot2 <- gen_PCoA_Num_Plot_2(data = beta.dt.full,
#                             x.var = "Body.Condition.Score",
#                             x.var.2 = "Timepoint",  # Optional
#                             y.var = methods.beta,
#                             elps.var = "Diet",  # Optional
#                             axis = beta.axis.full,
#                             stats = anova,  # Optional
#                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional
#                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
#                             disp.stats = T  # Optional
#                             )


plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "",
                        title = "",
                        colors = col.diet
                        )
plot3

```





```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2
plot3
# plot4

# Stats
stats_table_3(anova)

anova %>% pander()


beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Diet"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

```

\newpage

### Does body condition score affect microbiome differential abundance?

MaAsLin2 found no statistically significant effect of body condition score on differential abundance with random effects, but found significant associations when min abundance set to 0 (default) and with no random effects.


```{r diffAbund-bodyCond, echo=FALSE, message=FALSE, warning=FALSE}
# plot_diffAbund_2(DiffAbund_BodyCond_rEff[["Sig.Res"]] ) # No sig effects 
plot_diffAbund_2(DiffAbund_BodyCond[["Sig.Res"]] ) # No sig effects with random effects

```

```{r}
# DiffAbund_BodyCond[["Stats.Table"]]
```


```{r echo=FALSE}

test.highBC <- DiffAbund_BodyCondLvl[["Sig.Res"]] %>%
  filter(((value == ">= median") & (coef > 0)) | ((value == "6mpf") & (coef > 0)) ) %>%
  group_by(Genus) %>%
  count() %>%
  arrange(-n)

test.lowBC <- DiffAbund_BodyCondLvl[["Sig.Res"]] %>%
  filter(((value == ">= median") & (coef < 0)) | ((value == "6mpf") & (coef > 0)) ) %>%
  group_by(Genus) %>%
  count() %>%
  arrange(-n)


test.lowBC %>%
  filter(!Genus %in% test.highBC$Genus)

test.highBC %>%
  filter(!Genus %in% test.lowBC$Genus)

```


\newpage

### Does body condition score affect microbiome differential abundance depending on diet?

```{r diffAbund-bodyCond-Diet, echo=FALSE, fig.height=10, fig.width=6, message=FALSE, warning=FALSE}
plot_diffAbund_2(DiffAbund_DietBodyCond[["Sig.Res"]] )
# plot_diffAbund_2(DiffAbund_DietBodyCond_rEff[["Sig.Res"]] ) 

```

```{r}
# DiffAbund_DietBodyCond[["Stats.Table"]] 
```

\newpage

## Does variation in microbiome diversity/composition link to variation in infection status depend on diet?

### Does infection status affect microbiome diversity?


```{r Alpha-Infection, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.expFin.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Infection.Status",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})

plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
   gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Infection.Status",  # x-variable
                              "Alpha.Score",  # y-variable
                              # facet.var.x = "Sex",  # Optional
                              facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.infected,  # Optional
                              title = "glm(Alpha.Score ~ Infection.Status)"  # Optional
                              # title = NULL  # Optional
                              )
})



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

\newpage

### Does infection status affect microbiome diversity depending on diet?

```{r Alpha-Infection-Diet, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.expFin.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Infection.Status*Diet",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})



plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
   gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Infection.Status",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Diet",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.infected,  # Optional
                              title = "glm(Alpha.Score ~ Infection.Status*Diet)",  # Optional
                              # title = NULL  # Optional
                              y.lab = "Diversity (Shannon Entropy)",
                              comparisons = list( c("Absent", "Present")#, 
                                                  # c("Watts", "ZIRC"), 
                                                  # c("Gemma", "ZIRC") 
                                                  )
                              )
})

plot2 <- list()
plot2 <- lapply(names(methods.alpha), function(x){
  # subset(data, Alpha.Metric == x)
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Diet",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Infection.Status",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.infected,  # Optional
                              # title = "glm(Alpha.Score ~ Diet*Infection.Status)"  # Optional
                              title = ""  # Optional
                              )
})



```

```{r echo=FALSE, fig.height=5, fig.width=8}
# Plot
plot

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))

```

\newpage

### Does infection status affect microbiome diversity depending on diet in males?

```{r Alpha-Infection-Diet-males, message=FALSE, warning=FALSE, include=FALSE}

data <- subset(dt.alphaPlus.expFin.melt, Sex == "M")

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Infection.Status*Diet",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})


plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
   gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Infection.Status",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Diet",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.infected,  # Optional
                              title = "glm(Alpha.Score ~ Infection.Status*Diet)",  # Optional
                              # title = NULL  # Optional
                              y.lab = "Diversity (Shannon Entroy)"
                              )
})

plot2 <- list()
plot2 <- lapply(names(methods.alpha), function(x){
  # subset(data, Alpha.Metric == x)
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Diet",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Infection.Status",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.diet,  # Optional
                              title = "glm(Alpha.Score ~ Diet*Infection.Status)",  # Optional
                              # title = NULL  # Optional
                              y.lab = "Diversity (Shannon Entroy)"
                              )
})



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))

```

\newpage

### Does infection status affect microbiome composition?

```{r Beta-Infection, message=FALSE, warning=FALSE, include=FALSE}

data = df.expFin
ps.obj = ps.expFin
dist = distList.expFin


# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Infection.Status"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = distList.expFin,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 1,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Infection.Status",
                            # x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Infection.Status",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.infected
                            )

plot2 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "factor(Infection.Status)",
                        # x.var.2 = "Timepoint",
                        title = "",
                        colors = col.infected
                        )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# plot3
# plot4

# Stats
stats_table_3(anova)

anova %>% pander()

# How to run numerical variable on beta disp?
beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Infection.Status"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

```



\newpage


### Does infection status affect microbiome composition depending on diet?

```{r Beta-Infection-diet, message=FALSE, warning=FALSE, include=FALSE}

data = df.expFin
ps.obj = ps.expFin
dist = distList.expFin


# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Infection.Status", "Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)



# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Infection.Status",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.infected
                            )

plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Diet",
                            x.var.2 = "Infection.Status",  # Optional
                            y.var = methods.beta,
                            elps.var = "Infection.Status",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.diet
                            )

plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Infection.Status",
                        x.var.2 = "Diet",
                        title = "",
                        colors = col.infected
                        )

plot4 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "Infection.Status",
                        title = "",
                        colors = col.diet
                        )


```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
plot3
plot4

# Stats
stats_table_3(anova)

anova %>% pander()


beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Infection.Status"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = c("Diet"),
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F)

```

\newpage

### Does infection status affect microbiome differential abundance?


```{r diffAbund-inf, echo=FALSE, message=FALSE, warning=FALSE}

plot_diffAbund_2(DiffAbund_Infection[["Sig.Res"]] )
# plot_diffAbund_2(DiffAbund_Infection_rEff[["Sig.Res"]] ) # No sig effects with random effects

```

```{r}
# DiffAbund_DietBodyCond[["Stats.Table"]] 
```

\newpage

### Does infection status affect microbiome differential abundance depending on diet?

```{r diffAbund-inf-diet, echo=FALSE, fig.height=10, fig.width=6, message=FALSE, warning=FALSE}

plot_diffAbund_2(DiffAbund_DietInf[["Sig.Res"]] )
# plot_diffAbund_2(DiffAbund_DietInf_rEff[["Sig.Res"]] ) # No significant effects with random effects

```



\newpage

### Does severity score affect microbiome diversity?

```{r Alpha-Severity, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.expFin.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Severity.Score",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})


plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
    gen_scatter_plot_2(data = subset(data, Alpha.Metric == x),
                          "Alpha.Score",  # x-variable
                          "Severity.Score",  # y-variable
                          # facet.var.x = "Sex",  # Optional
                          facet.var.y = "Alpha.Metric",  # Optional
                          colors = col.severity,  # Optional
                          title = "glm(Alpha.Score ~ Severity.Score)"  # Optional
                          # title = NULL  # Optional
                          )  #+ theme(legend.position = "bottom")
})







```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))

```
\newpage

### Does severity score affect microbiome diversity depending on diet?

```{r Alpha-Severity-Diet, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.expFin.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Severity.Score*Diet",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})

plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
    gen_scatter_plot_2(data = subset(data, Alpha.Metric == x),
                          "Alpha.Score",  # x-variable
                          "Severity.Score",  # y-variable
                          facet.var.x = "Diet",  # Optional
                          facet.var.y = "Alpha.Metric",  # Optional
                          colors = col.severity,  # Optional
                          title = "glm(Alpha.Score ~ Severity.Score*Diet)"  # Optional
                          # title = NULL  # Optional
                          )  #+ theme(legend.position = "bottom")
})



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```


\newpage



### Does severity score affect microbiome composition?


```{r Beta-Severity, message=FALSE, warning=FALSE, include=FALSE}

data = df.expFin
ps.obj = ps.expFin
dist = distList.expFin

# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("factor(Severity.Score)"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 1,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "factor(Severity.Score)",
                            # x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "factor(Severity.Score)",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.severity
                            )

plot2 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "factor(Severity.Score)",
                        # x.var.2 = "Timepoint",
                        title = "",
                        colors = col.severity
                        )

```

```{r echo=FALSE, fig.height=5, fig.width=5}

# Plot
plot
plot2
# plot3
# plot4

# Stats
stats_table_3(anova)

anova %>% pander()

# How to run numerical variable on beta disp?
beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = "Severity.Score",
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F,
                factor = T)


```

\newpage



### Does severity score affect microbiome composition depending on diet?


```{r Beta-Severity-diet, message=FALSE, warning=FALSE, include=FALSE}

data = df.expFin
ps.obj = ps.expFin
dist = distList.expFin

# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("factor(Severity.Score)*Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 1,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "factor(Severity.Score)",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.severity
                            )

plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Diet",
                            x.var.2 = "factor(Severity.Score)",  # Optional
                            y.var = methods.beta,
                            elps.var = "factor(Severity.Score)",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.diet
                            )

plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "factor(Severity.Score)",
                        x.var.2 = "Diet",
                        title = "",
                        colors = col.severity
                        )

plot4 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "factor(Severity.Score)",
                        title = "",
                        colors = col.diet
                        )

```

```{r echo=FALSE, fig.height=5, fig.width=5}

# Plot
plot
plot2
plot3
plot4

# Stats
stats_table_3(anova)

anova %>% pander()

# How to run numerical variable on beta disp?
beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = "Severity.Score",
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F,
                factor = T)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = "Severity.Score",
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F,
                factor = F)

```

\newpage



### Does exposure status affect microbiome diversity?


```{r Alpha-Exposure, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Exposure",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})


plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
    gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Exposure",  # x-variable
                              "Alpha.Score",  # y-variable
                              # facet.var.x = "Diet",  # Optional
                              facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.exposed,  # Optional
                              title = "glm(Alpha.Score ~ Exposure)",  # Optional
                              # title = NULL  # Optional
                              y.lab = "Diversity (Shannon Entropy)"
                              ) + scale_x_discrete(label = c("Exp", "Unexp"))
})




```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```

\newpage

### Does exposure status affect microbiome diversity depend on diet?

```{r Alpha-Exposure-Diet, message=FALSE, warning=FALSE, include=FALSE}

data <- dt.alphaPlus.T1.melt

mod.list <- lapply(methods.alpha, function(alpha){
  glm( formula = "Alpha.Score ~ Exposure*Diet",
       data = subset(data, Alpha.Metric == alpha),
       family = "quasibinomial")
})


plot <- list()
plot <- lapply(names(methods.alpha), function(x){
   # subset(data, Alpha.Metric == x)
     gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Exposure",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Diet",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.exposed,  # Optional
                              title = "glm(Alpha.Score ~ Exposure*Diet)",  # Optional
                              # title = NULL  # Optional
                              y.lab = "Diversity (Shannon Entropy)",
                              comparisons = list( c("Unexp", "Exp")#, 
                                                  # c("Watts", "ZIRC"), 
                                                  # c("Gemma", "ZIRC") 
                                                  )
                              ) #+ scale_x_discrete(label = c("Exp", "Unexp"))
})

plot2 <- list()
plot2 <- lapply(names(methods.alpha), function(x){
  # subset(data, Alpha.Metric == x)
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Diet",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Exposure",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.diet,  # Optional
                              title = "glm(Alpha.Score ~ Diet*Exposure)",  # Optional
                              # title = NULL  # Optional
                              y.lab = "Diversity (Shannon Entropy)"
                              )
})


```

```{r echo=FALSE, fig.height=5, fig.width=8}
# Plot
plot
plot2

lapply(methods.alpha, function(x){
  mod.list[[x]] %>% summary() #%>% pander() #%>% report_table()
})



lapply(methods.alpha, function(x){
  mod.list[[x]] %>% Anova(type = 2)
})

stats_table_3( lapply(methods.alpha, function(x){
  gen_glm_anova(mod.list[[x]], x)
}) %>% bind_rows() ) %>% set_caption(caption = extract_modForm(mod.list[[1]]))
```


```{r echo=FALSE, fig.height=5, fig.width=8}

data <- dt.alphaPlus.all.melt

plot3 <- list()
plot3 <- lapply(names(methods.alpha), function(x){
  # subset(data, Alpha.Metric == x)
  gen_box_plot_2(data = subset(data, Alpha.Metric == x),
                              "Exposure",  # x-variable
                              "Alpha.Score",  # y-variable
                              facet.var.x = "Diet + Timepoint",  # Optional
                              # facet.var.y = "Alpha.Metric",  # Optional
                              colors = col.exposed,  # Optional
                              # title = "glm(Alpha.Score ~ Exposure*Diet)"  # Optional
                              # title = NULL  # Optional
                              y.lab = "Diversity (Shannon Entropy)", # Optional
                               comparisons = list( c("Unexp", "Exp")#, 
                                                                # c("Watts", "ZIRC"), 
                                                                # c("Gemma", "ZIRC") 
                                                                )
                              ) #+ scale_x_discrete(label = c("Exp", "Unexp"))
})

plot3
```



### Does exposure affect microbiome composition?


```{r Beta-exp, message=FALSE, warning=FALSE, include=FALSE}

data = df.T1
ps.obj = ps.T1
dist = distList.T1

# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Exposure"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 1,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Exposure",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Exposure",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.exposed
                            )

# plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            # x.var = "Diet",
                            # x.var.2 = "Exposure",  # Optional
                            # y.var = methods.beta,
                            # # elps.var = "Diet",  # Optional
                            # axis = beta.axis.full,
                            # stats = anova,  # Optional
                            # # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            # disp.stats = T,  # Optional
                            # colors = col.severity
                            # )

plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Exposure",
                        x.var.2 = "Diet",
                        title = "",
                        colors = col.exposed
                        )

# plot4 <- betaUncon_plot(data = data, 
#                         # method = "", 
#                         physeq = ps.obj, 
#                         # dist = "bray", 
#                         facet.var.x = ".", 
#                         facet.var.y = ".",
#                         x.var = "Diet",
#                         x.var.2 = "Exposure",
#                         title = ""
#                         )

```

```{r echo=FALSE, fig.height=5, fig.width=5}

# Plot
plot
# plot2
plot3
# plot4

# Stats
stats_table_3(anova)

anova %>% pander()

# How to run numerical variable on beta disp?
beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = "Exposure",
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F,
                factor = T)

# ps.obj <- get_intMatPSobj(physeq = ps.T1, 
#                                      data = dt.T1, 
#                                      vars = c("Diet", "Exposure"))
# 
# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = "Severity.Score",
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F,
#                 factor = T)
```

\newpage



### Does exposure affect microbiome composition depending on diet?


```{r Beta-exp-diet, message=FALSE, warning=FALSE, include=FALSE}

data = df.T1
ps.obj = ps.T1
dist = distList.T1

# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Exposure*Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 1,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)


# Plot
plot <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Exposure",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.exposed
                            )

plot2 <- gen_PCoA_Cat_Plot_2(data = beta.dt.full,
                            x.var = "Diet",
                            x.var.2 = "Exposure",  # Optional
                            y.var = methods.beta,
                            elps.var = "Exposure",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T,  # Optional
                            colors = col.diet
                            )

plot3 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Exposure",
                        x.var.2 = "Diet",
                        title = "",
                        colors = col.exposed
                        )

plot4 <- betaUncon_plot(data = data, 
                        # method = "", 
                        physeq = ps.obj, 
                        # dist = "bray", 
                        facet.var.x = ".", 
                        facet.var.y = ".",
                        x.var = "Diet",
                        x.var.2 = "Exposure",
                        title = "",
                        colors = col.diet
                        )

```




```{r echo=FALSE, fig.height=5, fig.width=5}

# Plot
plot
plot2
plot3
plot4

# Stats
stats_table_3(anova)

anova %>% pander()

# How to run numerical variable on beta disp?

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = "Exposure",
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F,
                factor = T)

beta_hom_disper(data = data,
                physeq = ps.obj,
                vars = "Diet",
                methods = c("bray",
                            "canberra",
                            "sor"
                            ),
                plot = F,
                factor = T)

```




### Does exposure affect microbiome differential abundance?


```{r diffAbund-exp, message=FALSE, warning=FALSE, include=FALSE}

plot_diffAbund_2(DiffAbund_Exposure[["Sig.Res"]] )
# plot_diffAbund_2(DiffAbund_Exposure_rEff[["Sig.Res"]] )  # No sig effects with random effects

# DiffAbund_DietTime[["Heatmap"]]



```

```{r}
test.Unexp <- DiffAbund_ExposureZIRC[["Sig.Res"]] %>%
  filter(coef > 0) %>%
  group_by(Genus) %>%
  count() %>%
  arrange(-n)

test.Exp <-DiffAbund_ExposureZIRC[["Sig.Res"]] %>%
  filter(coef < 0) %>%
  group_by(Genus) %>%
  count() %>%
  arrange(-n)

# Unexposed and Low Body Cond
test.Unexp %>%
  filter(!Genus %in% test.Exp$Genus & !Genus %in% test.highBC$Genus)

# Unexposed and High Body Cond
test.Unexp %>%
  filter(!Genus %in% test.Exp$Genus & !Genus %in% test.lowBC$Genus)

# Exp and Low Body Cond
test.Exp %>%
  filter(!Genus %in% test.Unexp$Genus & !Genus %in% test.highBC$Genus)

# Exp and High Body Cond
test.Exp %>%
  filter(!Genus %in% test.Unexp$Genus & !Genus %in% test.lowBC$Genus)
```



\newpage

### Does exposure affect microbiome differential abundance depending on diet?

```{r diffAbund-exp-diet, echo=FALSE, fig.height=10, fig.width=6, message=FALSE, warning=FALSE}

plot_diffAbund_2(DiffAbund_DietExp[["Sig.Res"]] )
# plot_diffAbund_2(DiffAbund_DietExp_rEff[["Sig.Res"]] )  # No sig effects with random effects

# DiffAbund_DietTime[["Heatmap"]]



```


```{r}
# DiffAbund_DietExp[["Stats.Table"]]
```
\newpage


# Does variation in microbiome diversity link to variation in microbiome composition?

## Alpha Diversity

### Does alpha diversity affect microbiome composition (initial)?


```{r Beta-alpha-Shan-initial, message=FALSE, warning=FALSE, include=FALSE}

data.alpha <- dt.alphaPlus.T0.melt
data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon")
data <- data.alpha
ps.obj <- ps.T0
dist <- distList.T0



# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Alpha.Score"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Merge alpha and beta dataframes
data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample")

# Plot
plot <- gen_PCoA_Num_Plot_2(data = data.merge,
                            x.var = "Alpha.Score",
                            # x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            title = "Beta.Score ~ Alpha.Score",  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )

# plot2 <- gen_PCoA_Num_Plot_2(data = data.merge,
#                             x.var = "Alpha.Score",
#                             x.var.2 = "Diet",  # Optional
#                             y.var = methods.beta,
#                             elps.var = "Sex",  # Optional
#                             axis = beta.axis.full,
#                             stats = anova,  # Optional
#                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional
#                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
#                             disp.stats = T  # Optional
#                             )


# plot3 <- betaUncon_plot(data = data.merge,
#                         # method = "",
#                         physeq = ps.obj,
#                         # dist = "bray",
#                         facet.var.x = ".",
#                         facet.var.y = ".",
#                         x.var = "Alpha.Score",
#                         # x.var.2 = "Timepoint",
#                         title = "",
#                         colors = col.diet
#                         )

# plot3 <- betaUncon_plot(data = data, 
#                         # method = "", 
#                         physeq = ps.obj, 
#                         # dist = "bray", 
#                         facet.var.x = ".", 
#                         facet.var.y = ".",
#                         x.var = "Body.Condition.Score",
#                         x.var.2 = "Timepoint",
#                         title = "",
#                         colors = col.diet
#                         ) 



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2
# plot3
# plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Diet"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)
# 
# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Timepoint"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)


```


\newpage

### Does alpha diversity affect microbiome composition (final)?


```{r Beta-alpha-Shan-final, message=FALSE, warning=FALSE, include=FALSE}
data.alpha <- dt.alphaPlus.T1.melt
data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon")
data <- data.alpha
ps.obj <- ps.T1
dist <- distList.T1



# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Alpha.Score"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Merge alpha and beta dataframes
data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample")

# Plot
plot <- gen_PCoA_Num_Plot_2(data = data.merge,
                            x.var = "Alpha.Score",
                            # x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            title = "Beta.Score ~ Alpha.Score",  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )

# plot2 <- gen_PCoA_Num_Plot_2(data = data.merge,
#                             x.var = "Alpha.Score",
#                             x.var.2 = "Diet",  # Optional
#                             y.var = methods.beta,
#                             elps.var = "Sex",  # Optional
#                             axis = beta.axis.full,
#                             stats = anova,  # Optional
#                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional
#                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
#                             disp.stats = T  # Optional
#                             )


# plot3 <- betaUncon_plot(data = data.merge,
#                         # method = "",
#                         physeq = ps.obj,
#                         # dist = "bray",
#                         facet.var.x = ".",
#                         facet.var.y = ".",
#                         x.var = "Alpha.Score",
#                         # x.var.2 = "Timepoint",
#                         title = "",
#                         colors = col.diet
#                         )

# plot3 <- betaUncon_plot(data = data, 
#                         # method = "", 
#                         physeq = ps.obj, 
#                         # dist = "bray", 
#                         facet.var.x = ".", 
#                         facet.var.y = ".",
#                         x.var = "Body.Condition.Score",
#                         x.var.2 = "Timepoint",
#                         title = "",
#                         colors = col.diet
#                         ) 



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2
# plot3
# plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Diet"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)
# 
# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Timepoint"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)


```


\newpage


## Diet

### Does alpha diversity affect microbiome composition depending on diet (initial)?


```{r Beta-alpha-Shan-Diet-initial, message=FALSE, warning=FALSE, include=FALSE}
data.alpha <- dt.alphaPlus.T0.melt
data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon")
data <- data.alpha

ps.obj <- ps.T0
dist <- distList.T0


# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Alpha.Score", "Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Merge alpha and beta dataframes
data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample")

# Plot
plot <- gen_PCoA_Num_Plot_2(data = data.merge,
                            x.var = "Alpha.Score",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            title = "Beta.Score ~ Alpha.Score + Diet",  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )

# plot2 <- gen_PCoA_Num_Plot_2(data = data.merge,
#                             x.var = "Alpha.Score",
#                             x.var.2 = "Diet",  # Optional
#                             y.var = methods.beta,
#                             elps.var = "Sex",  # Optional
#                             axis = beta.axis.full,
#                             stats = anova,  # Optional
#                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional
#                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
#                             disp.stats = T  # Optional
#                             )


# plot3 <- betaUncon_plot(data = data.merge,
#                         # method = "",
#                         physeq = ps.obj,
#                         # dist = "bray",
#                         facet.var.x = ".",
#                         facet.var.y = ".",
#                         x.var = "Alpha.Score",
#                         # x.var.2 = "Timepoint",
#                         title = "",
#                         colors = col.diet
#                         )

# plot3 <- betaUncon_plot(data = data, 
#                         # method = "", 
#                         physeq = ps.obj, 
#                         # dist = "bray", 
#                         facet.var.x = ".", 
#                         facet.var.y = ".",
#                         x.var = "Body.Condition.Score",
#                         x.var.2 = "Timepoint",
#                         title = "",
#                         colors = col.diet
#                         ) 



```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# plot3
# plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Diet"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)
# 
# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Timepoint"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)


```


\newpage


### Does alpha diversity affect microbiome composition depending on diet (final)?


```{r Beta-alpha-Shan-Diet-final, message=FALSE, warning=FALSE, include=FALSE}
data.alpha <- dt.alphaPlus.T1.melt
data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon")
data <- data.alpha

ps.obj <- ps.T1
dist <- distList.T1


# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Alpha.Score", "Diet"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Merge alpha and beta dataframes
data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample")

# Plot
plot <- gen_PCoA_Num_Plot_2(data = data.merge,
                            x.var = "Alpha.Score",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Diet",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            title = "Beta.Score ~ Alpha.Score + Diet",  # Optional
                            # title = "The effect of diet on the composition of the gut microbiome",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )




```

```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
plot2
# plot3
# plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Diet"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)
# 
# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Timepoint"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)


```


\newpage

## Time

### Does alpha diversity affect microbiome composition depending on time?


```{r Beta-alpha-Shan-time, message=FALSE, warning=FALSE, include=FALSE}

data.alpha <- dt.alphaPlus.conT0T1.melt
data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon")
data <- data.alpha

ps.obj <- ps.conT0T1
dist <- distList.conT0T1



# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Alpha.Score", "Timepoint"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Merge alpha and beta dataframes
data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample")

# Plot
plot <- gen_PCoA_Num_Plot_2(data = data.merge,
                            x.var = "Alpha.Score",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Timepoint",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            title = "Beta.Score ~ Alpha.Score + Timepoint",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )





```

```{r testing1, eval=FALSE, include=FALSE}
test.tmp.ord <- ordinate(
      physeq = ps.obj, 
      method = "PCoA",  # c("DCA", "CCA", "RDA", "CAP", "DPCoA", "NMDS", "MDS", "PCoA")
      distance = "bray"  # c("bray", "canberra", "sor")
    )

test.df <- subset(data.merge, Dist == "Bray-Curtis")

plot_ordination(
      physeq = ps.obj,
      ordination = test.tmp.ord,
    ) +
      geom_point(fill = test.df$Body.Condition.Score,
                     color = test.df$Body.Condition.Score,
                     # shape = test.df$Diet,
                     size = test.df$Body.Condition.Score,
                 alpha = .9) +
      # scale_size_area() +
  scale_color_distiller(palette = "Spectral") + theme(legend.position = "inside")
```






```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2
# plot3
# plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Diet"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)
# 
# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Timepoint"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)


```


\newpage

## Exposure

### Does alpha diversity affect microbiome composition depending on Exposure?


```{r Beta-alpha-Shan-exposure, message=FALSE, warning=FALSE, include=FALSE}

data.alpha <- dt.alphaPlus.T1.melt
data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon")
data <- data.alpha

ps.obj <- ps.T1
dist <- distList.T1



# Full dbRDA models
mod <- full_dbrda_model_2(vars = c("Alpha.Score", "Exposure"),  # Enter variables as a list "c(var.A, var.B, var.C)"
                          distance = dist,
                          methods = methods.beta,
                          names = set_names_beta,
                          physeq = ps.obj,
                          data = data,
                          terms = 2,  # if you want interaction terms enter 2, otherwise 1
                          num.cores = num.cores)

anova <- beta_anova(beta.model = mod,
                    methods = methods.beta,
                    names = set_names_beta,
                    num.cores = num.cores)


# Plot prep
dt.ord.full.beta <- ord_dt_list(mod, ps.obj)
beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta)
beta.axis.full <- axis_labels(dt.ord.full.beta)

# Merge alpha and beta dataframes
data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample")

# Plot
plot <- gen_PCoA_Num_Plot_2(data = data.merge,
                            x.var = "Alpha.Score",
                            x.var.2 = "Diet",  # Optional
                            y.var = methods.beta,
                            elps.var = "Exposure",  # Optional
                            axis = beta.axis.full,
                            stats = anova,  # Optional
                            title = "Beta.Score ~ Alpha.Score + Exposure",  # Optional
                            # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>"
                            disp.stats = T  # Optional
                            )

plot



```



```{r echo=FALSE, fig.height=5, fig.width=5}
# Plot
plot
# plot2
# plot3
# plot4

# Stats Table
stats_table_3(anova)

anova %>% pander()

# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Diet"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)
# 
# beta_hom_disper(data = data,
#                 physeq = ps.obj,
#                 vars = c("Timepoint"),
#                 methods = c("bray",
#                             "canberra",
#                             "sor"
#                             ),
#                 plot = F)


```


\newpage


## Body Condition Score

<!-- ### Does alpha diversity affect microbiome composition depending on body condition score (initial)? -->


<!-- ```{r Beta-alpha-Shan-body-initial, message=FALSE, warning=FALSE, include=FALSE} -->
<!-- data.alpha <- dt.alphaPlus.T0.melt -->
<!-- data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon") -->
<!-- data <- data.alpha -->

<!-- ps.obj <- ps.T0 -->
<!-- dist <- distList.T0 -->


<!-- # Full dbRDA models -->
<!-- mod <- full_dbrda_model_2(vars = c("Alpha.Score", "Body.Condition.Score"),  # Enter variables as a list "c(var.A, var.B, var.C)" -->
<!--                           distance = dist, -->
<!--                           methods = methods.beta, -->
<!--                           names = set_names_beta, -->
<!--                           physeq = ps.obj, -->
<!--                           data = data, -->
<!--                           terms = 2,  # if you want interaction terms enter 2, otherwise 1 -->
<!--                           num.cores = num.cores) -->

<!-- anova <- beta_anova(beta.model = mod, -->
<!--                     methods = methods.beta, -->
<!--                     names = set_names_beta, -->
<!--                     num.cores = num.cores) -->


<!-- # Plot prep -->
<!-- dt.ord.full.beta <- ord_dt_list(mod, ps.obj) -->
<!-- beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta) -->
<!-- beta.axis.full <- axis_labels(dt.ord.full.beta) -->

<!-- # Merge alpha and beta dataframes -->
<!-- data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample") -->

<!-- # Plot -->
<!-- plot <- gen_PCoA_Num_Plot_3(data = data.merge, -->
<!--                             x.var = "Alpha.Score", -->
<!--                             x.var.2 = "Body.Condition.Score",  # Optional -->
<!--                             y.var = methods.beta, -->
<!--                             elps.var = "Diet",  # Optional -->
<!--                             axis = beta.axis.full, -->
<!--                             stats = anova,  # Optional -->
<!--                             title = "Beta.Score ~ Alpha.Score + Body.Condition.Score",  # Optional -->
<!--                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional -->
<!--                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>" -->
<!--                             disp.stats = T  # Optional -->
<!--                             ) -->

<!-- plot2 <- gen_PCoA_Num_Plot_3(data = data.merge, -->
<!--                             x.var = "Body.Condition.Score", -->
<!--                             x.var.2 = "Alpha.Score",  # Optional -->
<!--                             y.var = methods.beta, -->
<!--                             elps.var = "Diet",  # Optional -->
<!--                             axis = beta.axis.full, -->
<!--                             stats = anova,  # Optional -->
<!--                             title = "Beta.Score ~ Alpha.Score + Body.Condition.Score",  # Optional -->
<!--                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional -->
<!--                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>" -->
<!--                             disp.stats = T  # Optional -->
<!--                             ) -->


<!-- ``` -->

<!-- ```{r echo=FALSE, fig.height=5, fig.width=5} -->
<!-- # Plot -->
<!-- plot -->
<!-- plot2 -->
<!-- # plot3 -->
<!-- # plot4 -->

<!-- # Stats Table -->
<!-- stats_table_3(anova) -->

<!-- anova %>% pander() -->

<!-- # beta_hom_disper(data = data, -->
<!-- #                 physeq = ps.obj, -->
<!-- #                 vars = c("Diet"), -->
<!-- #                 methods = c("bray", -->
<!-- #                             "canberra", -->
<!-- #                             "sor" -->
<!-- #                             ), -->
<!-- #                 plot = F) -->
<!-- #  -->
<!-- # beta_hom_disper(data = data, -->
<!-- #                 physeq = ps.obj, -->
<!-- #                 vars = c("Timepoint"), -->
<!-- #                 methods = c("bray", -->
<!-- #                             "canberra", -->
<!-- #                             "sor" -->
<!-- #                             ), -->
<!-- #                 plot = F) -->


<!-- ``` -->


<!-- \newpage -->


<!-- ### Does alpha diversity affect microbiome composition depending on body condition score (final)? -->


<!-- ```{r Beta-alpha-Shan-Diet-final2, message=FALSE, warning=FALSE, include=FALSE} -->
<!-- data.alpha <- dt.alphaPlus.T1.melt -->
<!-- data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon") -->
<!-- data <- data.alpha -->

<!-- ps.obj <- ps.T1 -->
<!-- dist <- distList.T1 -->


<!-- # Full dbRDA models -->
<!-- mod <- full_dbrda_model_2(vars = c("Alpha.Score", "Body.Condition.Score"),  # Enter variables as a list "c(var.A, var.B, var.C)" -->
<!--                           distance = dist, -->
<!--                           methods = methods.beta, -->
<!--                           names = set_names_beta, -->
<!--                           physeq = ps.obj, -->
<!--                           data = data, -->
<!--                           terms = 2,  # if you want interaction terms enter 2, otherwise 1 -->
<!--                           num.cores = num.cores) -->

<!-- anova <- beta_anova(beta.model = mod, -->
<!--                     methods = methods.beta, -->
<!--                     names = set_names_beta, -->
<!--                     num.cores = num.cores) -->


<!-- # Plot prep -->
<!-- dt.ord.full.beta <- ord_dt_list(mod, ps.obj) -->
<!-- beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta) -->
<!-- beta.axis.full <- axis_labels(dt.ord.full.beta) -->

<!-- # Merge alpha and beta dataframes -->
<!-- data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample") -->

<!-- # Plot -->
<!-- plot <- gen_PCoA_Num_Plot_3(data = data.merge, -->
<!--                             x.var = "Alpha.Score", -->
<!--                             x.var.2 = "Body.Condition.Score",  # Optional -->
<!--                             y.var = methods.beta, -->
<!--                             elps.var = "Diet",  # Optional -->
<!--                             axis = beta.axis.full, -->
<!--                             stats = anova,  # Optional -->
<!--                             title = "Beta.Score ~ Alpha.Score + Body.Condition.Score",  # Optional -->
<!--                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional -->
<!--                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>" -->
<!--                             disp.stats = T  # Optional -->
<!--                             ) -->

<!-- plot2 <- gen_PCoA_Num_Plot_3(data = data.merge, -->
<!--                             x.var = "Body.Condition.Score", -->
<!--                             x.var.2 = "Alpha.Score",  # Optional -->
<!--                             y.var = methods.beta, -->
<!--                             elps.var = "Diet",  # Optional -->
<!--                             axis = beta.axis.full, -->
<!--                             stats = anova,  # Optional -->
<!--                             title = "Beta.Score ~ Alpha.Score + Body.Condition.Score",  # Optional -->
<!--                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional -->
<!--                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>" -->
<!--                             disp.stats = T  # Optional -->
<!--                             ) -->





<!-- ``` -->

<!-- ```{r echo=FALSE, fig.height=5, fig.width=5} -->
<!-- # Plot -->
<!-- plot -->
<!-- plot2 -->
<!-- # plot3 -->
<!-- # plot4 -->

<!-- # Stats Table -->
<!-- stats_table_3(anova) -->

<!-- anova %>% pander() -->

<!-- # beta_hom_disper(data = data, -->
<!-- #                 physeq = ps.obj, -->
<!-- #                 vars = c("Diet"), -->
<!-- #                 methods = c("bray", -->
<!-- #                             "canberra", -->
<!-- #                             "sor" -->
<!-- #                             ), -->
<!-- #                 plot = F) -->
<!-- #  -->
<!-- # beta_hom_disper(data = data, -->
<!-- #                 physeq = ps.obj, -->
<!-- #                 vars = c("Timepoint"), -->
<!-- #                 methods = c("bray", -->
<!-- #                             "canberra", -->
<!-- #                             "sor" -->
<!-- #                             ), -->
<!-- #                 plot = F) -->


<!-- ``` -->


<!-- \newpage -->



<!-- ```{r} -->
<!-- data.alpha <- dt.alphaPlus.T1.melt -->
<!-- data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon") -->
<!-- data <- data.alpha -->

<!-- ps.obj <- ps.T1 -->
<!-- dist <- distList.T1 -->


<!-- # Full dbRDA models -->
<!-- mod <- full_dbrda_model_2(vars = c("Alpha.Score", "Body.Condition.Score"),  # Enter variables as a list "c(var.A, var.B, var.C)" -->
<!--                           distance = dist, -->
<!--                           methods = methods.beta, -->
<!--                           names = set_names_beta, -->
<!--                           physeq = ps.obj, -->
<!--                           data = data, -->
<!--                           terms = 2,  # if you want interaction terms enter 2, otherwise 1 -->
<!--                           num.cores = num.cores) -->

<!-- anova <- beta_anova(beta.model = mod, -->
<!--                     methods = methods.beta, -->
<!--                     names = set_names_beta, -->
<!--                     num.cores = num.cores) -->


<!-- # Plot prep -->
<!-- dt.ord.full.beta <- ord_dt_list(mod, ps.obj) -->
<!-- beta.dt.full <- samp_coord_dt(dt.ord.full.beta, methods.beta) -->
<!-- beta.axis.full <- axis_labels(dt.ord.full.beta) -->

<!-- # Merge alpha and beta dataframes -->
<!-- data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample") -->

<!-- # Plot -->
<!-- plot <- gen_PCoA_Num_Plot_3(data = data.merge, -->
<!--                             x.var = "Alpha.Score", -->
<!--                             x.var.2 = "Body.Condition.Score",  # Optional -->
<!--                             y.var = methods.beta, -->
<!--                             elps.var = "Diet",  # Optional -->
<!--                             axis = beta.axis.full, -->
<!--                             stats = anova,  # Optional -->
<!--                             title = "Beta.Score ~ Alpha.Score + Body.Condition.Score",  # Optional -->
<!--                             # title = "The effect of diet on the composition of the gut microbiome",  # Optional -->
<!--                             # title = NULL  # Null returns no title to the top, "" returns: "<beta.metric> ~ <variable>" -->
<!--                             disp.stats = T  # Optional -->
<!--                             ) -->


<!-- ``` -->








<!-- ```{r eval=FALSE, include=FALSE} -->
<!-- test.ps.obj <- ps.all -->
<!-- test.p.method <- "PCoA" -->
<!-- test.beta <- "bray" -->
<!-- test.tmp.p <- list() -->

<!-- test.data.alpha <- dt.alphaPlus.all.melt -->
<!-- test.data.alpha <- subset(data.alpha, Alpha.Metric == "Shannon") -->
<!-- test.data <- data.alpha -->
<!-- # test.data.merge <- merge(beta.dt.full, data.alpha[,c("Sample","Alpha.Score")], by = "Sample") -->
<!-- sample_data(test.ps.obj) <- data.frame(merge(sample.data.table(test.ps.obj), data.alpha[,c("Sample","Alpha.Score")], by = "Sample"), row.names=1) -->

<!-- test.tmp.ord <- ordinate( -->
<!--       physeq = test.ps.obj,  -->
<!--       method = test.p.method,  # c("DCA", "CCA", "RDA", "CAP", "DPCoA", "NMDS", "MDS", "PCoA") -->
<!--       distance = test.beta  # c("bray", "canberra", "sor") -->
<!--     ) -->





<!--     # Plot function -->
<!--     test.tmp.p[[test.beta]] <- plot_ordination( -->
<!--       physeq = test.ps.obj, -->
<!--       ordination = test.tmp.ord, -->
<!--     )  -->

<!--     test.p <- ggplot(data = test.tmp.p[[test.beta]]$data, aes(x= Axis.1, y = Alpha.Score)) +  -->
<!--       geom_point(aes(size = Body.Condition.Score, -->
<!--                      color = Alpha.Score, -->
<!--                      shape = Diet, -->
<!--                      fill = Diet)) +  -->
<!--       scale_colour_gradient( -->
<!--         low = pal.RdPu[3], -->
<!--         high = pal.paired[10], -->
<!--         space = "Lab", -->
<!--         na.value = "grey50", -->
<!--         guide = "colourbar", -->
<!--         aesthetics = "colour" -->
<!--       ) +  -->
<!--                       theme( -->
<!--                       # axis.line=element_blank(), -->
<!--                       # axis.text.x=element_blank(), -->
<!--                       # axis.text.y=element_blank(), -->
<!--                       # axis.ticks=element_blank(), -->
<!--                       # axis.title.x=element_blank(), -->
<!--                       # axis.title.y=element_blank(), -->
<!--                       legend.position="none" -->
<!--                       # panel.background=element_blank(), -->
<!--                       # panel.border=element_blank(), -->
<!--                       # panel.grid.major=element_blank(), -->
<!--                       # panel.grid.minor=element_blank(), -->
<!--                       # plot.background=element_blank() -->
<!--                       ) -->

<!--       ggMarginal(test.p, margins = "y", yparams = list(colour = pal.dark2[c(1,2,3)]), type = "boxplot", groupFill=T) -->

<!--       test.p.2 <- ggplot(data = test.tmp.p[[test.beta]]$data, aes(x= Diet, y = Alpha.Score)) +  -->
<!--                     geom_boxplot(aes(color = Alpha.Score, -->
<!--                                    fill = Diet)) + scale_fill_manual(values = pal.dark2) +  -->
<!--                       theme( -->
<!--                       axis.line=element_blank(), -->
<!--                       # axis.text.x=element_blank(), -->
<!--                       axis.text.y=element_blank(), -->
<!--                       # axis.ticks=element_blank(), -->
<!--                       # axis.title.x=element_blank(), -->
<!--                       axis.title.y=element_blank(), -->
<!--                       legend.position="none" -->
<!--                       # panel.background=element_blank(), -->
<!--                       # panel.border=element_blank(), -->
<!--                       # panel.grid.major=element_blank(), -->
<!--                       # panel.grid.minor=element_blank(), -->
<!--                       # plot.background=element_blank() -->
<!--                       ) -->

<!--       # test.p.3 <- ggplot(data = test.tmp.p[[test.beta]]$data, aes(x= Diet, y = Body.Condition.Score)) +  -->
<!--       #               geom_boxplot(aes( -->
<!--       #                 color = Body.Condition.Score)) +  -->
<!--       #                 scale_fill_distiller(palette = "Spectrum") +  -->
<!--       #                 theme( -->
<!--       #                 axis.line=element_blank(), -->
<!--       #                 # axis.text.x=element_blank(), -->
<!--       #                 axis.text.y=element_blank(), -->
<!--       #                 # axis.ticks=element_blank(), -->
<!--       #                 # axis.title.x=element_blank(), -->
<!--       #                 axis.title.y=element_blank(), -->
<!--       #                 legend.position="none" -->
<!--       #                 # panel.background=element_blank(), -->
<!--       #                 # panel.border=element_blank(), -->
<!--       #                 # panel.grid.major=element_blank(), -->
<!--       #                 # panel.grid.minor=element_blank(), -->
<!--       #                 # plot.background=element_blank() -->
<!--       #                 ) -->



<!--       plot_grid(test.p, test.p.2, nrow = 1, align = "v", rel_widths = c(1, .5))   -->





<!--     #+ -->
<!--       geom_point(aes(fill = eval(parse(text = x.var)), -->
<!--                      color = eval(parse(text = x.var)), -->
<!--                      shape = eval(parse(text = x.var.2))), -->
<!--                  alpha = 0.7, size = 4) + -->
<!--       stat_ellipse(aes(linetype = eval(parse(text = tmp.ellipse)) )) + -->
<!--       # geom_label_repel(data = calc_Centroid_2(tmp.ord, x.var = x.var), -->
<!--       #                  aes(label = eval(parse(text = x.var)) ), -->
<!--       #                  alpha = 0.80) + -->
<!--       facet_grid(paste0(facet.var.y, " ~ ", facet.var.x), scales = "free_y") + -->
<!--       # scale_color_brewer(palette = "Dark2") + -->
<!--       # scale_fill_brewer(palette = "Dark2") + -->
<!--       scale_fill_manual(values = colors) + -->
<!--       scale_color_manual(values = colors) + -->
<!--       labs( -->
<!--         # title = ifelse(title != "", title, paste0(c(p.method, beta), collapse = ", ")) -->
<!--         title = ifelse(title == "", paste0("Beta.Score ~ ", ifelse(x.var.2 == "", x.var, paste0(c(x.var, x.var.2), collapse = " + ")) ), title), -->
<!--         caption = paste0(c(p.method, beta), collapse = ", ") -->
<!--       ) + -->
<!--       guides(fill = guide_legend(title=x.var), -->
<!--              color = guide_legend(title=x.var),#, override.aes = list(size=3)), -->
<!--              shape = guide_legend(title=x.var.2), -->
<!--              #override.aes = list(alpha = 1, size = 3) ), -->
<!--              linetype=guide_legend(title="Ellipses"), -->
<!--              size = F, -->
<!--              alpha = F) -->
<!-- ``` -->

