---
title: "Zebrafish Diet and Infection Targetted Analysis"
author: "Connor Draney, Sunni Patton, Kelly Shannon, Michael Sieler"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document:
    toc: yes
    toc_depth: '3'
  html_document:
    toc: yes
    toc_depth: 3
    number_sections: yes
    theme: united
    highlight: tango
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# Check project file location is in correct directory
## Path to project directory (Rproject file should be stored in "~/Code")
proj.path <- file.path(paste0(getwd(),"/../Code"))


# Helper Scripts

## Load libraries
source(paste0(proj.path,"/HelperScripts/libraries.R"))

## Load functions
source(paste0(proj.path,"/HelperScripts/Functions.R")) 

## Load custom plot settings
source(paste0(proj.path,"/HelperScripts/plot_settings.R"))


# Set important paths

## Path to data files/sub-folders
data.path <- paste0(proj.path,  # Project path
                               "/../Data")  # Sub-directory

## Path to Robjects (Renv, Rds, Rdata, etc.) saved here
objects.path <- paste0(data.path,  # Project path
                                  "/Clean/Robjects")  # Sub-directory

## Path to Output (Figures, Tables)
output.path <- paste0(proj.path,  # Project path
                                 "/../Output")  # Sub-directory


```


# Import Data

We obtained data in the form of a [phyloseq](https://joey711.github.io/phyloseq/) data object from the [Human Genome Project](https://www.hmpdacc.org/hmp/) (HMP) that was previously processed for taxanomic identification of microbial organisms. We then cleaned and processed the data further for statistical analysis.

```{r import-data}

# Download Data
# temp_test = tempfile()
# test_url = "http://joey711.github.io/phyloseq-demo/HMPv35.RData"
# download.file(test_url, destfile = paste0(data.path, "/Raw/HMPv35.RData"))

load(paste0(data.path, "/Raw/HMPv35.RData"))
ps.unclean <- HMPv35
rm(HMPv35)  # remove this obj from global env
```

## Clean sample data

The 4743 samples in the HMP were obtained from multiple individuals (2555 males and 2188 females), processed by several locations, and in some cases repeated samples were taken from the same individual. We subset the data to ensure independence by limiting samples from one location and only the first visit. Additionally, we removed samples noted as "Mislabeled" or "Contaminated". After removing samples using these criteria, we were left with 609 samples.


```{r clean-sample-data}

# Find the center with largest counts for the following conditions
sample.data.frame(ps.unclean) %>% 
  count(RUNCENTER,
        Mislabeled,
        Contaminated, 
        visitno) %>% 
  arrange(-n)

# Subset based on parameters
ps.cleaned <- subset_samples(ps.unclean,  # physeq object
                              Mislabeled == F &  # Remove mislabeled
                              Contaminated == F &  # remove contaminated
                              visitno == 1 & # first visit, so no repeat sampling of individuals, independence
                              RUNCENTER == "JCVI"  # Same center, independence
                              )


# Remove columns, if rownames are already sample names, no need to have an extra sample column
sample_data(ps.cleaned) <- sample_data(ps.cleaned)[, c(-1)]  # [rows, cols]

# Check to see how many samples dropped from original dataset
print(paste0("Dropped ", 
             nrow(sample_data(ps.unclean)) - nrow(sample_data(ps.cleaned)), 
             " samples after cleaning.")
            )

# head(sample.data.frame(ps.cleaned))


```



# Rarefaction

We rarefied the data to control for uneven sampling efforts  [Sanders, H. L. (1968)](https://www.journals.uchicago.edu/doi/abs/10.1086/282541), [Willis, A.D.  (2019)](https://www.frontiersin.org/articles/10.3389/fmicb.2019.02407/full#B13).

```{r rarefaction, message=FALSE, warning=FALSE}
ps.norm <- ps.cleaned
rarefaction.minimum <- 1000
min.smpl.size <- min(sample_sums(ps.norm)[sample_sums(ps.norm) >= rarefaction.minimum])

summary(sample_sums(ps.norm))

# Rarefy

ps.rar <- {
  ps.rar.tmp0 <- ps.norm
  ps.rar.tmp1 <- rarefy_even_depth(
      physeq = ps.rar.tmp0, 
      sample.size = min.smpl.size,
      trimOTUs = TRUE,
      rngseed = 42
      )
  rename.NA.taxa(ps.rar.tmp1)
}

ps.rar

```




# Plot pre- and post-rarefaction

Check the number of reads before and after rarefaction.

```{r pre-post-norm}
# rare.plot.preRare <- rarecurve(t(otu_table(ps.norm)), step=50, cex=0.5, label=F)
hist(sample_sums(ps.norm), main="Histogram: Read Counts before rarefaction", xlab="Total Reads",
     border="blue", col="green", las=1, breaks=12)
hist(sample_sums(ps.rar), main="Histogram: Read Counts after rarefaction", xlab="Total Reads",
     border="blue", col="green", las=1, breaks=12)

```



```{r physeq-obj-list}
# Make a list of phyloseq objects, dataframes and datatables
ps.list <- list(RAR = list(ps.all = ps.rar,
                           df.all = sample.data.frame(ps.rar),
                           dt.all = sample.data.table(ps.rar) 
                           ))

# Add observation numbers to data
ps.list[["RAR"]][["df.all"]] <- dplyr::mutate(ps.list[["RAR"]][["df.all"]], obs_num = row_number(), .before = 1)
ps.list[["RAR"]][["dt.all"]] <- dplyr::mutate(ps.list[["RAR"]][["dt.all"]], obs_num = row_number(), .before = 1)


# head(ps.list$RAR$df.all)


```


# Alpha-diversity

We calculated alpha diversity scores, a measure of number of unique organisms within a single sample, using the indices Shannon, Simpson, Phylogenetic, and Richness (Whittaker 1960). Each calculates alpha-diversity using slightly different mathematical approaches to measure evenness (distribution of organisms) and/or richness (number of organismss). If differences in results are seen between indices, this can reveal insights into which kinds of organisms are present (e.g., common vs rare).

```{r alpha-diversity}
# Select which alpha measures we want to analyze
methods.alpha <- c("Shannon", "Simpson", # Non-phylogenetic measures, add additional measures here
                   "Phylogenetic", "Richness") %>%   
                  purrr::set_names()  # Set's names list elements in "alpha.methods"

# Calculate alpha scores, save to list
ps.list$RAR[["alphaScore"]] <- alpha_base(ps.list$RAR$ps.all,  # Phyloseq object
                               methods.alpha,  # list of alpha methods
                               "Sample",
                               phylo.div = T
                               )


# Add alpha scores to data table
ps.list$RAR[["dt.all.alpha"]] <- ps.list$RAR$dt.all[ps.list$RAR$alpha, on = "Sample"] %>% setkeyv("Sample")

# Melt data table
ps.list$RAR[["dt.all.alpha.melt"]] <- melt_to_datatable_2(datatable1 = ps.list$RAR$dt.all, 
                                                               datatable2 = ps.list$RAR$dt.all.alpha, 
                                                               vars = methods.alpha, 
                                                               var.name = "Alpha.Metric",
                                                               samp.name = "Sample",
                                                               val.name = "Alpha.Score"
                                                              )


```


# Fit linear model

We fit the data using linear model to predict alpha-diversity score as a function of body sub-site, sex, or their interaction.

```{r fit-lm}
caseInfStats <- list()

data <- ps.list$RAR[["dt.all.alpha.melt"]]
# data <- dplyr::mutate(data, obs_num = row_number(), .before = 1)

caseInfStats[["mod.unref"]] <- lapply(methods.alpha, function(alpha){
  lm( formula = "Alpha.Score ~ sex*HMPbodysubsite",
       data = subset(data, Alpha.Metric == alpha)
      )
})

```

## Check assumptions (unrefined model)

To assess if the assumptions of linear model regression were met, we visually inspected residuals vs fitted values, standardized residuals vs quantiles, standardized residuals vs fitted values and residuals vs leverage for each diversity index. In general, Residuals vs fitted appears slightly heteroscedasticity, Q-Q plots show that the data slightly deviates from the diagonal line indicating that the data may be non-normal, Scale-location plots show that std. residuals are negatively associated with fitted values indicating heteroscedasticity, and there are several points with high leverage, but none that appear to have too high. The plots associated with the Simpson's index appear to be significantly heteroscedastic and non-normal.


### Shannon

```{r}

index <- "Shannon"
plot(caseInfStats[["mod.unref"]][[index]], sub = paste0("(Unrefined: ", methods.alpha[index],")") )

```

### Simpson



```{r}

index <- "Simpson"
plot(caseInfStats[["mod.unref"]][[index]], sub = paste0("(Unrefined: ", methods.alpha[index],")") )

```

### Phylogenetic



```{r}

index <- "Phylogenetic"
plot(caseInfStats[["mod.unref"]][[index]], sub = paste0("(Unrefined: ", methods.alpha[index],")") )

```

### Richness



```{r}

index <- "Richness"
plot(caseInfStats[["mod.unref"]][[index]], sub = paste0("(Unrefined: ", methods.alpha[index],")") )

```




## Remove unusual observations

The previous plots revealed that the data is largely consistent, but there are some inconsistencies in variance and non-normality for some of the alpha indices. To correct for this, we applied case influence statistics to remove any unusually influential and leveraged samples. We removed any samples using cutoffs of 2p/n (p is the number of unknown parameters in the model) for leverage, 1 for Cook’s Distance and above 2 or below -2 for standardized residuals.



```{r check-assumptions, echo = T, results = 'hide', message=FALSE, warning=FALSE}

caseInfStats <- list()

data <- ps.list$RAR[["dt.all.alpha.melt"]]
# data <- dplyr::mutate(data, obs_num = row_number(), .before = 1)

caseInfStats[["mod.unref"]] <- lapply(methods.alpha, function(alpha){
  lm( formula = "Alpha.Score ~ sex*HMPbodysubsite",
       data = subset(data, Alpha.Metric == alpha)
      )
})

# Fortify data for plotting
caseInfStats[["dataFort.unref"]] <- lapply(methods.alpha, function(alpha){
  fortify(caseInfStats$mod.unref[[alpha]], subset(data, Alpha.Metric == alpha))
})

# Rename some column names
lapply(methods.alpha, function(alpha){
  setnames(caseInfStats$dataFort.unref[[alpha]], 
           old=c(".hat", ".cooksd", ".stdresid"), 
           new=c("Lev", "CooksD", "StdResid"))
})


# Unrefinded model

## Plot

caseInfStats[["unref.plot"]] <- lapply(names(methods.alpha), function(alpha){
  qplot(obs_num, value, data = reshape::melt(caseInfStats$dataFort.unref[[alpha]][, c("obs_num","Lev", "CooksD", "StdResid")],
    id.vars = "obs_num")) + 
    geom_point(aes(color = variable)) +
    facet_grid(variable ~ ., scale = "free_y") + 
    labs(title = paste0("Case-influence statistics plot: Unrefined model (", alpha,")")) + 
    scale_color_brewer(palette = "Dark2") + 
    theme(legend.position = "none") + scale_x_continuous(breaks = scales::breaks_pretty(10))

}) 

names(caseInfStats[["unref.plot"]]) <- names(methods.alpha)



```

From the original 562 samples, we removed 45, 54, 33, and 31 unusual observations in the Shannon, Simpson, Phylogenetic and Richness indices, respectively.

# Fit refined model

After removing the unusual samples, we refit the data to a linear model.

```{r refined-model, echo = T, results = 'hide', message=FALSE, warning=FALSE}

# Refined model

### Statisticians use rough cutoffs of 2p/n (p is the number of unknown parameters in the model) for leverage, 
#     1 for Cook’s Distance and above 2 or below -2 for standardized residuals. 
#     Observations falling outside these ranges warrant further attention.

caseInfStats[["cutoff.lev"]] <- lapply(methods.alpha, function(alpha){
  cutOff.lev <- (2 * length(caseInfStats$mod.unref[[alpha]][["coefficients"]]) / 
                   nrow(caseInfStats$dataFort.unref[[alpha]]))
})


caseInfStats[["dataFort.sub"]] <- lapply(methods.alpha, function(alpha){
  caseInfStats$dataFort.unref[[alpha]]  %>%
    dplyr::filter(Lev < caseInfStats[["cutoff.lev"]][[alpha]]) %>%  # Leverage Cut off
    dplyr::filter(StdResid < 2 & StdResid > -2 ) %>%  # StdResid Cut Off
    dplyr::select(obs_num:Alpha.Score)  # Removes the old case statistic influence data
})


# make refined model
caseInfStats[["mod.ref"]] <- lapply(methods.alpha, function(alpha){
  lm( formula = "Alpha.Score ~ sex*HMPbodysubsite",
       data = caseInfStats[["dataFort.sub"]][[alpha]]
      )
})


# Fortify data for plotting
caseInfStats[["dataFort.ref"]] <- lapply(methods.alpha, function(alpha){
  fortify(caseInfStats$mod.ref[[alpha]], caseInfStats[["dataFort.sub"]][[alpha]])
})

# Rename some column names
lapply(methods.alpha, function(alpha){
  setnames(caseInfStats$dataFort.ref[[alpha]], 
           old=c(".hat", ".cooksd", ".stdresid"), 
           new=c("Lev", "CooksD", "StdResid"))
})


# Refinded model Plot

caseInfStats[["ref.plot"]] <- lapply(names(methods.alpha), function(alpha){
  tmp.data <- caseInfStats[["dataFort.ref"]]
  qplot(obs_num, value, data = reshape::melt(tmp.data[[alpha]][, c("obs_num","Lev", "CooksD", "StdResid")],
    id.vars = "obs_num")) + 
    geom_point(aes(color = variable)) +
    facet_grid(variable ~ ., scale = "free_y") + 
    labs(title = paste0("Case-influence statistics plot: Refined model (", alpha,")")) + 
    scale_color_brewer(palette = "Dark2") + 
    theme(legend.position = "none") + scale_x_continuous(breaks = scales::breaks_pretty(10))

}) 

names(caseInfStats[["ref.plot"]]) <- names(methods.alpha)

```



## Check assumptions (refined model)

After removing the unusual observations, the plots for each indice appear to have been improved. However, the Simpson's data appears to not meet the assumptions of normality.

### Shannon
```{r}

index <- "Shannon"
plot(caseInfStats[["mod.ref"]][[index]], sub = paste0("(Refined: ", methods.alpha[index],")") )

```

### Simpson
```{r}

index <- "Simpson"
plot(caseInfStats[["mod.ref"]][[index]], sub = paste0("(Refined: ", methods.alpha[index],")") )

```

### Phylogenetic
```{r}

index <- "Phylogenetic"
plot(caseInfStats[["mod.ref"]][[index]], sub = paste0("(Refined: ", methods.alpha[index],")") )

```

### Richness
```{r}

index <- "Richness"
plot(caseInfStats[["mod.ref"]][[index]], sub = paste0("(Refined: ", methods.alpha[index],")") )

```




## Plot unrefined and refined models



```{r plot-unref-ref, echo=FALSE, message=FALSE, warning=FALSE}

# Combine Plots
caseStatisticsPlots <- lapply(methods.alpha, function(alpha){
   marrangeGrob(list(caseInfStats[["unref.plot"]][[alpha]], 
                     caseInfStats[["ref.plot"]][[alpha]]), 
                nrow = 2, ncol = 1, top=NULL)
})


caseStatisticsPlots

```


# Linear model results

## Shannon

```{r message=FALSE, warning=FALSE}
index <- "Shannon"

Anova(caseInfStats[["mod.ref"]][[index]], type = 2)
caseInfStats[["mod.ref"]][[index]] %>% report()
Anova(caseInfStats[["mod.ref"]][[index]], type = 2) %>% report()

```

## Simpson

```{r message=FALSE, warning=FALSE}
index <- "Simpson"

Anova(caseInfStats[["mod.ref"]][[index]], type = 2)
caseInfStats[["mod.ref"]][[index]] %>% report()
Anova(caseInfStats[["mod.ref"]][[index]], type = 2) %>% report()

```

## Phylogenetic

```{r message=FALSE, warning=FALSE}
index <- "Phylogenetic"

Anova(caseInfStats[["mod.ref"]][[index]], type = 2)
caseInfStats[["mod.ref"]][[index]] %>% report()
Anova(caseInfStats[["mod.ref"]][[index]], type = 2) %>% report()

```

## Richness

```{r message=FALSE, warning=FALSE}
index <- "Richness"

Anova(caseInfStats[["mod.ref"]][[index]], type = 2)
caseInfStats[["mod.ref"]][[index]] %>% report()
Anova(caseInfStats[["mod.ref"]][[index]], type = 2) %>% report()

```






# Plots

```{r plot-unnorm}


ps.list[["RAR"]][["Plots"]] <- list()
ps.list[["RAR"]][["Plots"]] <- lapply(names(methods.alpha), function(alpha){
  tmp.data <- na.omit(as.data.frame(caseInfStats[["dataFort.ref"]][[alpha]]) )
  ggplot(tmp.data, aes(x = HMPbodysubsite, y=Alpha.Score)) +
      geom_boxplot(aes(fill = HMPbodysubsite)) + 
      ggbeeswarm::geom_quasirandom(size = 0.75) +  # spaces the dots out nicely
      facet_grid(. ~ .) +  # (Y-axis ~ X-axis)
      theme(legend.position = "none",
            axis.text.x = element_text(angle = 33, hjust = 1, vjust=1)
            ) +
      labs(
        title = "Diversity Scores by Body Sub-Site",
        # caption = "",
        y = paste0("Diversity (", alpha,")"),
        x = "Body Sub-Site"
      ) 
}) 

names(ps.list[["RAR"]][["Plots"]]) <- names(methods.alpha)

ps.list[["RAR"]][["Plots"]]

```



